<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Pedro L. Baldoni" />


<title>Analysis of the RNA-seq data from the epithelial cell population of the mouse mammary gland</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TranscriptDE-wf</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/plbaldoni/wf-TranscriptDE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of the RNA-seq data from the
epithelial cell population of the mouse mammary gland</h1>
<h4 class="author">Pedro L. Baldoni</h4>
<h4 class="date">17 February, 2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-02-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>TranscriptDE-wf/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221115code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221115)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221115code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221115)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomplbaldoniTranscriptDEwftree21fff0174c41654438796c73c4cd4f0588118e49targetblank21fff01a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/tree/21fff0174c41654438796c73c4cd4f0588118e49" target="_blank">21fff01</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomplbaldoniTranscriptDEwftree21fff0174c41654438796c73c4cd4f0588118e49targetblank21fff01a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/plbaldoni/TranscriptDE-wf/tree/21fff0174c41654438796c73c4cd4f0588118e49" target="_blank">21fff01</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    .gitignore
    Ignored:    analysis/simulation-complete_cache/
    Ignored:    analysis/simulation-paper_cache/
    Ignored:    code/mouse/single-end/salmon/slurm-9574761.out
    Ignored:    code/pkg/.Rhistory
    Ignored:    code/pkg/.Rproj.user/
    Ignored:    code/pkg/pkg.Rproj
    Ignored:    code/pkg/src/RcppExports.o
    Ignored:    code/pkg/src/pkg.so
    Ignored:    code/pkg/src/rcpparma_hello_world.o
    Ignored:    data/annotation/mm39/
    Ignored:    data/mouse/paired-end/fastq/
    Ignored:    data/mouse/single-end/fastq/
    Ignored:    misc/.DS_Store
    Ignored:    misc/._.DS_Store
    Ignored:    misc/mouse.Rmd/._figure6.png
    Ignored:    misc/simulation-paper.Rmd/._figure2.png
    Ignored:    misc/simulation-paper.Rmd/._figure5.png
    Ignored:    output/mouse/paired-end/
    Ignored:    output/mouse/single-end/
    Ignored:    output/quasi_poisson/
    Ignored:    output/simulation/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/mouse.Rmd</code>) and HTML
(<code>docs/mouse.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/21fff0174c41654438796c73c4cd4f0588118e49/analysis/mouse.Rmd" target="_blank">21fff01</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Organizing mouse report
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/b8e397902f76d2cf942e92fb1eb38fffc7ef1eb3/docs/mouse.html" target="_blank">b8e3979</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/57b0d0081f0113c473a272fb1b8ef921d68ee788/analysis/mouse.Rmd" target="_blank">57b0d00</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Renaming repo and organizing main page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/57b0d0081f0113c473a272fb1b8ef921d68ee788/docs/mouse.html" target="_blank">57b0d00</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Renaming repo and organizing main page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/e75e3d99d906c0c30ba538acfb3516f8fce9547d/analysis/mouse.Rmd" target="_blank">e75e3d9</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Adjusting figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/e75e3d99d906c0c30ba538acfb3516f8fce9547d/docs/mouse.html" target="_blank">e75e3d9</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Adjusting figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/edbcc15a55c43c524a8282264f42e3eedf65d538/analysis/mouse.Rmd" target="_blank">edbcc15</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Repeat header
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/753efade4b9b9460503137657e8c03c922fe2230/analysis/mouse.Rmd" target="_blank">753efad</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Adjusting latex tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/623d4297221b6b8a0e0e44292e19626bf6ed3270/analysis/mouse.Rmd" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Splitting figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/623d4297221b6b8a0e0e44292e19626bf6ed3270/docs/mouse.html" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Splitting figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/e64cf4d4c222149ac5cade052a4feb1ee5325e47/analysis/mouse.Rmd" target="_blank">e64cf4d</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-19
</td>
<td>
Changing values of reference to report in main paper
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/e64cf4d4c222149ac5cade052a4feb1ee5325e47/docs/mouse.html" target="_blank">e64cf4d</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-19
</td>
<td>
Changing values of reference to report in main paper
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/9a3e9261caa134d79e86c10b792aee7e08f08ff5/analysis/mouse.Rmd" target="_blank">9a3e926</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-06
</td>
<td>
Organizing output from mouse analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/9a3e9261caa134d79e86c10b792aee7e08f08ff5/docs/mouse.html" target="_blank">9a3e926</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-06
</td>
<td>
Organizing output from mouse analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/c228d3f26a2f820f85d2baa971ed77a8f554d689/analysis/mouse.Rmd" target="_blank">c228d3f</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-05
</td>
<td>
Updating mouse report
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/c228d3f26a2f820f85d2baa971ed77a8f554d689/docs/mouse.html" target="_blank">c228d3f</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-05
</td>
<td>
Updating mouse report
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/481735cfb080507aebd58b49f552aea110fa7ed5/docs/mouse.html" target="_blank">481735c</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-24
</td>
<td>
Build update from workflowr
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/3e9c510e1bf819c729c3058d6c149ee6fa1ad1de/analysis/mouse.Rmd" target="_blank">3e9c510</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-22
</td>
<td>
Adding mouse report
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-wf/5ee11169d27eee64b3ddaf34ea11cd4a6645871a/docs/mouse.html" target="_blank">5ee1116</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-22
</td>
<td>
Updating docs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this page, we present the analysis of the epithelial cell
population of the mouse mammary gland. The report is divided in two main
parts: the analysis of the paired-end data from the basal, luminal
progenitor (LP), and mature luminal (ML) cells, and the analysis of the
single-end data from the basal and LP cells (<a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60450">GSE60450</a>).</p>
<p>The analysis presented in this report begins with the experimental
data already quantified by <code>Salmon</code>. Please refer to the
script files located in the GitHub repository of this page under the
directory path <code>./code/mouse</code> which show the commands used to
quantify the RNA-seq reads from these experiments. The
<code>targets</code> files from these experiments are located in
<code>./data/mouse</code>.</p>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>We begin this report by setting up some options for rendering this
webpage and loading the necessary libraries.</p>
<pre class="r"><code>knitr::opts_chunk$set(
  dev = &quot;png&quot;,
  dpi = 300,
  dev.args = list(type = &quot;cairo-png&quot;),
  root.dir = &#39;.&#39;
)</code></pre>
<pre class="r"><code>library(edgeR)
library(data.table)
library(ggplot2)
library(readr)
library(Rsubread)
library(rtracklayer)
library(magrittr)
library(ggpubr)
library(plyr)
library(gplots)
library(grid)
library(ComplexHeatmap)
library(patchwork)
library(tibble)
library(tidyHeatmap)
library(AnnotationHub)
library(sleuth)
library(fishpond)
library(tximeta)
library(tidyverse)
library(SummarizedExperiment)
library(stringr)
library(ragg)
library(kableExtra)</code></pre>
<p>Below we load the Ensembl 104 annotation from
<code>AnnotationHub</code> which is associated to the Ensembl M27
annotation of the mouse mm39 genome.</p>
<pre class="r"><code># &#39;AH95775&#39; annotation corresponds to Ensembl 104 (release M27)
ah &lt;- AnnotationHub()</code></pre>
<pre><code>snapshotDate(): 2022-10-31</code></pre>
<pre class="r"><code>edb &lt;- ah[[&#39;AH95775&#39;]]</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre class="r"><code>ensid &lt;- keys(edb)
cols &lt;- c(&quot;TXIDVERSION&quot;,&quot;TXEXTERNALNAME&quot;,&quot;TXBIOTYPE&quot;,&quot;GENEIDVERSION&quot;,
          &quot;GENENAME&quot;,&quot;GENEBIOTYPE&quot;,&quot;ENTREZID&quot;)

dt.anno &lt;- select(edb,ensid,cols)
dt.anno &lt;- as.data.table(dt.anno)

dt.anno.gene &lt;- 
  dt.anno[,.(NTranscriptPerGene = .N),
          by = c(&#39;GENEIDVERSION&#39;,&#39;GENEBIOTYPE&#39;,&#39;ENTREZID&#39;,&#39;GENENAME&#39;)]
dt.anno.gene[,GeneOfInterest := GENEBIOTYPE %in% c(&#39;protein_coding&#39;,&#39;lncRNA&#39;)]

dt.anno &lt;- 
  merge(dt.anno,dt.anno.gene[,-c(2,3,4)],by = &#39;GENEIDVERSION&#39;,all.x = TRUE)</code></pre>
<p>Paths used in this report are specified below.</p>
<pre class="r"><code>path.anno &lt;- &#39;../data/annotation/mm39&#39;

path.misc &lt;- file.path(&#39;../misc&#39;,knitr::current_input())
dir.create(path.misc,recursive = TRUE,showWarnings = FALSE)

path.data.pe &lt;- &#39;../data/mouse/paired-end&#39;
path.quant.pe &lt;- &#39;../output/mouse/paired-end&#39;

path.data.se &lt;- &#39;../data/mouse/single-end&#39;
path.quant.se &lt;- &#39;../output/mouse/single-end&#39;</code></pre>
</div>
</div>
<div id="analysis-of-paired-end-data" class="section level1">
<h1>Analysis of paired-end data</h1>
<p>The analysis of paired-end data is divided in two parts. First, we
present the analysis at the transcript-level. Second, we present a
gene-level analysis. All results discussed in the main paper regarding
these experiments are presented in this report.</p>
<div id="transcript-level-analysis" class="section level2">
<h2>Transcript-level analysis</h2>
<div id="data-wrangling" class="section level3">
<h3>Data wrangling</h3>
<p>The transcript-level analysis starts with some data wrangling.</p>
<pre class="r"><code>dt.targets.pe &lt;- fread(file.path(path.data.pe,&#39;misc/targets.txt&#39;))
dt.targets.pe[,Sample := paste(Group,Replicate,sep = &#39;.&#39;)]
dt.targets.pe[,Color := mapvalues(Group,
                                  from = c(&#39;Basal&#39;,&#39;LP&#39;,&#39;ML&#39;),
                                  to = c(&#39;blue&#39;,&#39;darkgreen&#39;,&#39;red&#39;))]
setnames(dt.targets.pe,old = &#39;Group&#39;,new = &#39;group&#39;)
dt.targets.pe[,path := 
                file.path(path.quant.pe,&#39;salmon&#39;,gsub(&#39;_R1.fastq.gz&#39;,&#39;&#39;,File1))]</code></pre>
<p>We then load <code>Salmon</code>’s output with the function
<code>catchSalmon</code> from <code>edgeR</code> while bringing in some
information about the Ensembl M27 annotation.</p>
<pre class="r"><code>catch.pe &lt;- catchSalmon(dt.targets.pe$path,verbose = FALSE)

key.catch.pe.anno &lt;- match(rownames(catch.pe$annotation),dt.anno$TXIDVERSION)

catch.pe$annotation$TranscriptName &lt;- dt.anno$TXEXTERNALNAME[key.catch.pe.anno]
catch.pe$annotation$GeneID &lt;- dt.anno$GENEIDVERSION[key.catch.pe.anno]
catch.pe$annotation$GeneName &lt;- dt.anno$GENENAME[key.catch.pe.anno]
catch.pe$annotation$GeneEntrezID &lt;- dt.anno$ENTREZID[key.catch.pe.anno]
catch.pe$annotation$GeneOfInterest &lt;- dt.anno$GeneOfInterest[key.catch.pe.anno]
catch.pe$annotation$NTranscriptPerGene &lt;- dt.anno$NTranscriptPerGene[key.catch.pe.anno]
catch.pe$annotation$Type &lt;- dt.anno$TXBIOTYPE[key.catch.pe.anno]</code></pre>
</div>
<div id="differential-transcript-expression" class="section level3">
<h3>Differential transcript expression</h3>
<div id="edger-with-count-scaling" class="section level4">
<h4>edgeR with count scaling</h4>
<p>First, we assess DTE with the <code>edgeR</code> via count scaling
pipeline.</p>
<pre class="r"><code>dte.pe.scaled &lt;- 
  DGEList(counts = catch.pe$counts/catch.pe$annotation$Overdispersion,
          genes = catch.pe$annotation,
          samples = dt.targets.pe)
colnames(dte.pe.scaled) &lt;- dte.pe.scaled$samples$Sample</code></pre>
<p>Some basic filtering is done below. We restrict our analyses to
protein-coding genes and lncRNA.</p>
<pre class="r"><code>keep.pe.scaled &lt;- 
  filterByExpr(dte.pe.scaled) &amp; dte.pe.scaled$genes$GeneOfInterest
dte.pe.scaled.filtr &lt;- dte.pe.scaled[keep.pe.scaled,, keep.lib.sizes = FALSE]</code></pre>
<p>Next, we specify the design matrix of our experiment, calculate
normalizing factors and estimate the trended negative binomial
dispersions.</p>
<pre class="r"><code>design.pe &lt;- model.matrix(~0+group,data = dte.pe.scaled.filtr$samples)
dte.pe.scaled.filtr &lt;- calcNormFactors(dte.pe.scaled.filtr)
dte.pe.scaled.filtr &lt;- estimateDisp(dte.pe.scaled.filtr,design.pe,robust = TRUE)</code></pre>
<p>We proceed with the QL pipeline from <code>edgeR</code> by first
calling <code>glmQLFit</code> and then running QL tests with
<code>glmQLFTest</code>.</p>
<pre class="r"><code>fit.pe.scaled &lt;- glmQLFit(dte.pe.scaled.filtr,design.pe,robust = TRUE)

con.LPvsB &lt;- makeContrasts(LPvsB = groupLP - groupBasal,levels = design.pe)
con.MLvsLP &lt;- makeContrasts(MLvsLP = groupML - groupLP,levels = design.pe)

qlf.pe.LPvsB.scaled &lt;- glmQLFTest(fit.pe.scaled,contrast = con.LPvsB)
qlf.pe.MLvsLP.scaled &lt;- glmQLFTest(fit.pe.scaled,contrast = con.MLvsLP)

out.pe.LPvsB.scaled &lt;- topTags(qlf.pe.LPvsB.scaled,n = Inf)
out.pe.MLvsLP.scaled &lt;- topTags(qlf.pe.MLvsLP.scaled,n = Inf)

summary(decideTests(qlf.pe.LPvsB.scaled))</code></pre>
<pre><code>       -1*groupBasal 1*groupLP
Down                      9002
NotSig                   26192
Up                        8382</code></pre>
<pre class="r"><code>summary(decideTests(qlf.pe.MLvsLP.scaled))</code></pre>
<pre><code>       -1*groupLP 1*groupML
Down                   2422
NotSig                38639
Up                     2515</code></pre>
</div>
<div id="edger-with-raw-counts" class="section level4">
<h4>edgeR with raw counts</h4>
<p>Next, we run the <code>edgeR</code> pipeline with raw counts without
deflating counts by their associated mapping ambiguity overdispersion.
The pipeline is similar to the one used with scaled counts, and the only
difference is regarding count deflation which is now absent.</p>
<pre class="r"><code>dte.pe.raw &lt;- DGEList(counts = catch.pe$counts,
                      genes = catch.pe$annotation,
                      samples = dt.targets.pe)
colnames(dte.pe.raw) &lt;- dte.pe.raw$samples$Sample</code></pre>
<pre class="r"><code>keep.pe.raw &lt;- 
  filterByExpr(dte.pe.raw) &amp; dte.pe.raw$genes$GeneOfInterest
dte.pe.raw.filtr &lt;- dte.pe.raw[keep.pe.raw,, keep.lib.sizes = FALSE]</code></pre>
<pre class="r"><code>dte.pe.raw.filtr &lt;- calcNormFactors(dte.pe.raw.filtr)
dte.pe.raw.filtr &lt;- estimateDisp(dte.pe.raw.filtr,design.pe,robust = TRUE)</code></pre>
<pre class="r"><code>fit.pe.raw &lt;- glmQLFit(dte.pe.raw.filtr,design.pe,robust = TRUE)

qlf.pe.LPvsB.raw &lt;- glmQLFTest(fit.pe.raw,contrast = con.LPvsB)
qlf.pe.MLvsLP.raw &lt;- glmQLFTest(fit.pe.raw,contrast = con.MLvsLP)

out.pe.LPvsB.raw &lt;- topTags(qlf.pe.LPvsB.raw,n = Inf)
out.pe.MLvsLP.raw &lt;- topTags(qlf.pe.MLvsLP.raw,n = Inf)

summary(decideTests(qlf.pe.LPvsB.raw))</code></pre>
<pre><code>       -1*groupBasal 1*groupLP
Down                      8216
NotSig                   40699
Up                        7322</code></pre>
<pre class="r"><code>summary(decideTests(qlf.pe.MLvsLP.raw))</code></pre>
<pre><code>       -1*groupLP 1*groupML
Down                    673
NotSig                54922
Up                      642</code></pre>
</div>
<div id="sleuth-lrt" class="section level4">
<h4>sleuth-LRT</h4>
<p>We now run <code>sleuth</code> with LRT.</p>
<pre class="r"><code>dt.targets.pe.sleuth &lt;- dt.targets.pe[group %in% c(&#39;Basal&#39;,&#39;LP&#39;),]
setnames(dt.targets.pe.sleuth,old = &#39;Sample&#39;,new = &#39;sample&#39;)

se.pe.sleuth.lrt &lt;- 
  sleuth_prep(sample_to_covariates = dt.targets.pe.sleuth,full_model = ~ group)</code></pre>
<pre><code>Warning in check_num_cores(num_cores): It appears that you are running Sleuth from within Rstudio.
Because of concerns with forking processes from a GUI, &#39;num_cores&#39; is being set to 1.
If you wish to take advantage of multiple cores, please consider running sleuth from the command line.</code></pre>
<pre><code>reading in kallisto results</code></pre>
<pre><code>dropping unused factor levels</code></pre>
<pre><code>......
normalizing est_counts
58890 targets passed the filter
normalizing tpm
merging in metadata
summarizing bootstraps
......</code></pre>
<pre class="r"><code>se.pe.sleuth.lrt &lt;- sleuth_fit(obj = se.pe.sleuth.lrt, fit_name = &#39;full&#39;)</code></pre>
<pre><code>fitting measurement error models
shrinkage estimation
1 NA values were found during variance shrinkage estimation due to mean observation values outside of the range used for the LOESS fit.
The LOESS fit will be repeated using exact computation of the fitted surface to extrapolate the missing values.
These are the target ids with NA values: ENSMUST00000042235.15
computing variance of betas</code></pre>
<pre class="r"><code>se.pe.sleuth.lrt &lt;- 
  sleuth_fit(obj = se.pe.sleuth.lrt,formula = ~ 1,fit_name = &#39;reduced&#39;)</code></pre>
<pre><code>fitting measurement error models
shrinkage estimation
1 NA values were found during variance shrinkage estimation due to mean observation values outside of the range used for the LOESS fit.
The LOESS fit will be repeated using exact computation of the fitted surface to extrapolate the missing values.
These are the target ids with NA values: ENSMUST00000230860.2
computing variance of betas</code></pre>
<pre class="r"><code>se.pe.sleuth.lrt &lt;- 
  sleuth_lrt(obj = se.pe.sleuth.lrt,null_model = &#39;reduced&#39;,alt_model = &#39;full&#39;)

out.pe.sleuth.lrt &lt;- 
  sleuth_results(obj = se.pe.sleuth.lrt,
                 test = &#39;reduced:full&#39;, test_type = &#39;lrt&#39;,show_all = FALSE)</code></pre>
</div>
<div id="sleuth-wald" class="section level4">
<h4>sleuth-Wald</h4>
<p>We now run <code>sleuth</code> with Wald tests.</p>
<pre class="r"><code>se.pe.sleuth.wald &lt;- 
  sleuth_prep(sample_to_covariates = dt.targets.pe.sleuth,full_model = ~ group)</code></pre>
<pre><code>Warning in check_num_cores(num_cores): It appears that you are running Sleuth from within Rstudio.
Because of concerns with forking processes from a GUI, &#39;num_cores&#39; is being set to 1.
If you wish to take advantage of multiple cores, please consider running sleuth from the command line.</code></pre>
<pre><code>reading in kallisto results</code></pre>
<pre><code>dropping unused factor levels</code></pre>
<pre><code>......
normalizing est_counts
58890 targets passed the filter
normalizing tpm
merging in metadata
summarizing bootstraps
......</code></pre>
<pre class="r"><code>se.pe.sleuth.wald &lt;- sleuth_fit(obj = se.pe.sleuth.wald, fit_name = &#39;full&#39;)</code></pre>
<pre><code>fitting measurement error models
shrinkage estimation
1 NA values were found during variance shrinkage estimation due to mean observation values outside of the range used for the LOESS fit.
The LOESS fit will be repeated using exact computation of the fitted surface to extrapolate the missing values.
These are the target ids with NA values: ENSMUST00000042235.15
computing variance of betas</code></pre>
<pre class="r"><code>se.pe.sleuth.wald &lt;- 
  sleuth_wt(obj = se.pe.sleuth.wald,which_beta = &#39;groupLP&#39;,which_model = &#39;full&#39;)

out.pe.sleuth.wald &lt;- 
  sleuth_results(obj = se.pe.sleuth.wald, test = &#39;groupLP&#39;, test_type = &#39;wald&#39;,
                 show_all = FALSE)</code></pre>
</div>
<div id="swish" class="section level4">
<h4>Swish</h4>
<p>Finally, we run <code>Swish</code>.</p>
<pre class="r"><code>dt.targets.pe.swish &lt;- dt.targets.pe[group %in% c(&#39;Basal&#39;,&#39;LP&#39;),]
dt.targets.pe.swish[,files := file.path(path,&#39;quant.sf&#39;)]
dt.targets.pe.swish$group %&lt;&gt;% factor(levels = c(&#39;Basal&#39;,&#39;LP&#39;))
setnames(dt.targets.pe.swish,old = &#39;Sample&#39;,new = &#39;names&#39;)

se.pe.swish &lt;- tximeta(coldata = dt.targets.pe.swish,type = &#39;salmon&#39;)</code></pre>
<pre><code>importing quantifications</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 
found matching transcriptome:
[ GENCODE - Mus musculus - release M27 ]
loading existing TxDb created: 2022-04-05 23:03:51
loading existing transcript ranges created: 2022-04-05 23:03:52
fetching genome info for GENCODE</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 87 out-of-bound ranges located on sequences
  chr4, chr8, chr13, chr14, and chr17. Note that ranges located on a
  sequence whose length is unknown (NA) or on a circular sequence are not
  considered out-of-bound (use seqlengths() and isCircular() to get the
  lengths and circularity flags of the underlying sequences). You can use
  trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more
  information.</code></pre>
<pre class="r"><code>se.pe.swish &lt;- scaleInfReps(se.pe.swish)
se.pe.swish &lt;- labelKeep(se.pe.swish)
se.pe.swish &lt;- se.pe.swish[mcols(se.pe.swish)$keep,]
se.pe.swish &lt;- swish(y = se.pe.swish, x = &quot;group&quot;)
out.pe.swish &lt;- as.data.frame(mcols(se.pe.swish))</code></pre>
</div>
</div>
</div>
<div id="differential-gene-expression" class="section level2">
<h2>Differential gene expression</h2>
<p>Now, we perform a gene-level analysis of the same data.</p>
<p>The function below computes estimates the mapping ambiguity
overdispersion parameter at the level of gene-wise counts. It implements
the exact same formula from <code>catchSalmon</code>, but it instead
uses the aggregated counts at the gene-level from
<code>tximport::summarizeToGene</code>.</p>
<pre class="r"><code># Function computing catchSalmon&#39;s formula for gene-level counts
# To be used only for exploratory purposes
geneLevelCatchSalmon &lt;- function(x) {
  NSamples &lt;- ncol(x)
  NBoot &lt;- sum(grepl(&#39;infRep&#39;, assayNames(x)))
  NTx &lt;- nrow(x)
  DF &lt;- rep_len(0L, NTx)
  OverDisp &lt;- rep_len(0, NTx)
  
  for (i.samples in 1:NSamples) {
    Boot &lt;- lapply(1:NBoot, function(i.boot) {
      assay(x, paste0(&#39;infRep&#39;, i.boot))[, i.samples]
    })
    Boot &lt;- do.call(cbind, Boot)
    M &lt;- rowMeans(Boot)
    i &lt;- (M &gt; 0)
    OverDisp[i] &lt;- OverDisp[i] + rowSums((Boot[i,] - M[i]) ^ 2) / M[i]
    DF[i] &lt;- DF[i] + NBoot - 1L
  }
  
  i &lt;- (DF &gt; 0L)
  OverDisp[i] &lt;- OverDisp[i] / DF[i]
  DFMedian &lt;- median(DF[i])
  DFPrior &lt;- 3
  OverDispPrior &lt;-
    median(OverDisp[i]) / qf(0.5, df1 = DFMedian, df2 = DFPrior)
  if (OverDispPrior &lt; 1) {
    OverDispPrior &lt;- 1
  }
  OverDisp[i] &lt;-
    (DFPrior * OverDispPrior + DF[i] * OverDisp[i]) / (DFPrior + DF[i])
  OverDisp &lt;- pmax(OverDisp, 1)
  OverDisp[!i] &lt;- OverDispPrior
  rowData(x)$Overdispersion &lt;- OverDisp
  return(x)
}</code></pre>
<p>We summarize counts to the gene-level below and run the standard
<code>edgeR</code> QL pipeline.</p>
<pre class="r"><code>dt.targets.pe.tximeta &lt;- dt.targets.pe
dt.targets.pe.tximeta[,files := file.path(path,&#39;quant.sf&#39;)]
dt.targets.pe.tximeta$group %&lt;&gt;% factor(levels = c(&#39;Basal&#39;,&#39;LP&#39;,&#39;ML&#39;))
setnames(dt.targets.pe.tximeta,old = &#39;Sample&#39;,new = &#39;names&#39;)

txm &lt;- tximeta(coldata = dt.targets.pe.tximeta[,c(&#39;files&#39;,&#39;names&#39;)])</code></pre>
<pre><code>importing quantifications</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 
found matching transcriptome:
[ GENCODE - Mus musculus - release M27 ]
loading existing TxDb created: 2022-04-05 23:03:51
loading existing transcript ranges created: 2022-04-05 23:03:52
fetching genome info for GENCODE</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 87 out-of-bound ranges located on sequences
  chr4, chr8, chr13, chr14, and chr17. Note that ranges located on a
  sequence whose length is unknown (NA) or on a circular sequence are not
  considered out-of-bound (use seqlengths() and isCircular() to get the
  lengths and circularity flags of the underlying sequences). You can use
  trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more
  information.</code></pre>
<pre class="r"><code>se.pe.gene &lt;- summarizeToGene(txm)</code></pre>
<pre><code>loading existing TxDb created: 2022-04-05 23:03:51
obtaining transcript-to-gene mapping from database
loading existing gene ranges created: 2022-06-02 01:59:12</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 46 out-of-bound ranges located on sequences
  chr14, chr8, chr17, chr4, and chr13. Note that ranges located on a
  sequence whose length is unknown (NA) or on a circular sequence are not
  considered out-of-bound (use seqlengths() and isCircular() to get the
  lengths and circularity flags of the underlying sequences). You can use
  trim() to trim these ranges. See ?`trim,GenomicRanges-method` for more
  information.</code></pre>
<pre><code>summarizing abundance
summarizing counts
summarizing length
summarizing inferential replicates</code></pre>
<pre class="r"><code>se.pe.gene &lt;- geneLevelCatchSalmon(se.pe.gene)

dge.pe &lt;- DGEList(counts = assay(se.pe.gene,&#39;counts&#39;), 
                  samples = dt.targets.pe.tximeta,
                  genes = as.data.frame(rowData(se.pe.gene)))

key.dge.pe.anno.gene &lt;- match(rownames(dge.pe),dt.anno.gene$GENEIDVERSION)

dge.pe$genes$GeneOfInterest &lt;- dt.anno.gene$GeneOfInterest[key.dge.pe.anno.gene]
dge.pe$genes$NTranscriptPerGene &lt;- dt.anno.gene$NTranscriptPerGene[key.dge.pe.anno.gene]
dge.pe$genes$GeneEntrezID &lt;- dt.anno.gene$ENTREZID[key.dge.pe.anno.gene]
dge.pe$genes$GeneName &lt;- dt.anno.gene$GENENAME[key.dge.pe.anno.gene]
dge.pe$genes$GeneType &lt;- dt.anno.gene$GENEBIOTYPE[key.dge.pe.anno.gene]

keep.pe &lt;- filterByExpr(dge.pe) &amp; dge.pe$genes$GeneOfInterest

dge.pe.filtr &lt;- dge.pe[keep.pe, , keep.lib.sizes = FALSE]
dge.pe.filtr &lt;- calcNormFactors(dge.pe.filtr)

dge.pe.filtr &lt;-
  estimateDisp(dge.pe.filtr, design = design.pe, robust = TRUE)

fit.pe &lt;- glmQLFit(dge.pe.filtr,design.pe, robust = TRUE)

qlf.pe.LPvsB &lt;- glmQLFTest(fit.pe,contrast = con.LPvsB)
qlf.pe.MLvsLP &lt;- glmQLFTest(fit.pe,contrast = con.MLvsLP)

out.pe.LPvsB &lt;- topTags(qlf.pe.LPvsB,n = Inf) 
out.pe.MLvsLP &lt;- topTags(qlf.pe.MLvsLP,n = Inf) 

summary(decideTests(qlf.pe.LPvsB))</code></pre>
<pre><code>       -1*groupBasal 1*groupLP
Down                      4735
NotSig                    8488
Up                        4388</code></pre>
<pre class="r"><code>summary(decideTests(qlf.pe.MLvsLP))</code></pre>
<pre><code>       -1*groupLP 1*groupML
Down                   1512
NotSig                14381
Up                     1718</code></pre>
</div>
<div id="plots-and-other-exploratory-analyses" class="section level2">
<h2>Plots and other exploratory analyses</h2>
<p>Here we generate a few plots. We begin by assessing the mapping
ambiguity overdispersion parameter of each transcript (y-axis) according
the number of expressed transcripts associated with their parent gene
(x-axis). This is Figure 1 of the main paper.</p>
<pre class="r"><code>dt.mao.plot &lt;- as.data.table(dte.pe.scaled.filtr$genes)
dt.mao.plot[,NTranscriptPerGeneTrunc := ifelse(NTranscriptPerGene&lt;10,NTranscriptPerGene,paste0(&#39;&gt;=10&#39;))]

dt.mao.plot$NTranscriptPerGeneTrunc %&lt;&gt;% factor(levels = paste0(c(1:10,&#39;&gt;=10&#39;)))

# Number of transcripts from single-transcript genes
dt.mao.plot[NTranscriptPerGene == 1,
            .(.N,sum(NTranscriptPerGene == 1 &amp; Overdispersion&gt;(1/0.9))/sum(NTranscriptPerGene == 1))]</code></pre>
<pre><code>      N        V2
1: 2967 0.1681834</code></pre>
<pre class="r"><code># Number of transcripts from multi-transcript genes
dt.mao.plot[NTranscriptPerGene &gt; 1,
            .(.N,sum(NTranscriptPerGene &gt; 1 &amp; Overdispersion&gt;(1/0.9))/sum(NTranscriptPerGene &gt; 1))]</code></pre>
<pre><code>       N        V2
1: 40609 0.8127755</code></pre>
<pre class="r"><code># Number of transcripts from transcript-rich genes (#tx&gt;10)
dt.mao.plot[NTranscriptPerGene &gt;= 10,
            .(NGeneID = length(unique(GeneID)),
              NTranscriptID = .N,mean(Overdispersion))]</code></pre>
<pre><code>   NGeneID NTranscriptID       V3
1:    2172         12558 5.315015</code></pre>
<pre class="r"><code>plot.mao.2 &lt;- 
  ggplot(data = dt.mao.plot,aes(x = NTranscriptPerGeneTrunc,y = log10(Overdispersion))) +
  geom_boxplot(fill = &#39;#bebebe&#39;,outlier.alpha = 0.2,col = &#39;black&#39;) +
  geom_smooth(aes(group = 1),color = &#39;#ff0000&#39;,se = FALSE,span = 0.8,method = &#39;loess&#39;) +
  labs(x = &#39;Number of expressed transcripts per gene&#39;, y = &#39;Overdispersion (log10 scale)&#39;) +
  scale_y_continuous(limits = c(0,3),breaks = seq(0,3,0.5)) +
  theme_bw(base_size = 8,base_family = &#39;sans&#39;) +
  theme(panel.grid = element_blank(),
        axis.text = element_text(colour = &#39;black&#39;,size = 8))

agg_png(filename = file.path(path.misc,&quot;figure1.png&quot;),width = 5,height = 5,units = &#39;in&#39;,res = 300)
plot.mao.2</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>plot.mao.2</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/mouse.Rmd/mouse_eda_plots_mao-1.png" width="1500" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-mouse_eda_plots_mao-1">
Past versions of mouse_eda_plots_mao-1.png
</button>
</p>
<div id="fig-mouse_eda_plots_mao-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/623d4297221b6b8a0e0e44292e19626bf6ed3270/docs/figure/mouse.Rmd/mouse_eda_plots_mao-1.png" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/9a3e9261caa134d79e86c10b792aee7e08f08ff5/docs/figure/mouse.Rmd/mouse_eda_plots_mao-1.png" target="_blank">9a3e926</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Then, we generate Figure 6 of the main paper below.</p>
<pre class="r"><code># Heatmap of Foxp1 transcripts
cpm.scaled.filtr &lt;- cpm(dte.pe.scaled.filtr,log = TRUE)
rownames(cpm.scaled.filtr) &lt;- dte.pe.scaled.filtr$genes$TranscriptName

cpm.scaled.filtr.LPvsBasal &lt;- 
  cpm.scaled.filtr[,grepl(&#39;Basal|LP&#39;,colnames(cpm.scaled.filtr))]

cpm.scaled.filtr.LPvsBasal &lt;- t(scale(t(cpm.scaled.filtr.LPvsBasal)))

cpm.scaled.filtr.LPvsBasal.foxp1 &lt;- 
  cpm.scaled.filtr.LPvsBasal[grepl(&#39;foxp1&#39;,rownames(cpm.scaled.filtr.LPvsBasal),ignore.case = TRUE),]

foo.heat &lt;- function(x,cluster_rows = TRUE, fontsize = 8){
  tb &lt;- data.table(TranscriptName = rownames(x),x)
  tb &lt;- melt(tb,id.vars = &#39;TranscriptName&#39;,value.name = &#39;Expression&#39;)
  tb &lt;- as_tibble(tb)
  
  tb %&gt;%
    heatmap(.row = TranscriptName,.column = variable,.value = Expression,
            scale = &#39;none&#39;,palette_value = c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;),
            column_title = NULL,row_title = NULL,
            cluster_rows = cluster_rows,
            row_names_gp = gpar(fontsize = fontsize),
            column_names_gp = gpar(fontsize = fontsize),
            clustering_method_rows = &quot;complete&quot;,
            clustering_method_columns = &quot;complete&quot;,
            heatmap_legend_param = list(direction = &quot;horizontal&quot;,
                                        at = seq(-3,3,1),
                                        title_position = &#39;topcenter&#39;,
                                        title_gp = gpar(fontsize = fontsize),
                                        labels_gp = gpar(fontsize = fontsize))) %&gt;%
    as_ComplexHeatmap() %&gt;%
    draw(heatmap_legend_side = &quot;top&quot;)
}

plot.heat &lt;- 
  wrap_elements(grid.grabExpr(draw(foo.heat(cpm.scaled.filtr.LPvsBasal.foxp1))))</code></pre>
<pre><code>tidyHeatmap says: (once per session) from release 1.7.0 the scaling is set to &quot;none&quot; by default. Please use scale = &quot;row&quot;, &quot;column&quot; or &quot;both&quot; to apply scaling</code></pre>
<pre class="r"><code># Heatmap of significant transcripts of non-significant breast cancer genes
tb.pe.LPvsB.scaled &lt;- as.data.table(out.pe.LPvsB.scaled$table)
tb.pe.LPvsB &lt;- as.data.table(out.pe.LPvsB$table)

tb.pe.LPvsB.scaled$FDR.Gene &lt;- 
  tb.pe.LPvsB$FDR[match(tb.pe.LPvsB.scaled$GeneID,tb.pe.LPvsB$gene_id)]
tb.pe.LPvsB.scaled$logFC.Gene &lt;- 
  tb.pe.LPvsB$logFC[match(tb.pe.LPvsB.scaled$GeneID,tb.pe.LPvsB$gene_id)]

interestGenes &lt;- 
  tb.pe.LPvsB.scaled[FDR.Gene &gt; 0.05 &amp; 
                       FDR &lt; 0.05 &amp;
                       !is.na(GeneEntrezID),unique(GeneEntrezID)]

# Number of non-significant genes for which at least one of their transcripts is DE
length(interestGenes)</code></pre>
<pre><code>[1] 1818</code></pre>
<pre class="r"><code># Running KEGG analysis
tb.kegga &lt;- kegga(interestGenes,species = &#39;Mm&#39;)
GK &lt;- getGeneKEGGLinks(species.KEGG = &quot;mmu&quot;)
tb.kegga[&#39;path:mmu05224&#39;,]</code></pre>
<pre><code>                    Pathway   N DE      P.DE
path:mmu05224 Breast cancer 147 15 0.2533814</code></pre>
<pre class="r"><code>interestGenes.breast &lt;- 
  interestGenes[interestGenes %in% GK$GeneID[GK$PathwayID == &#39;path:mmu05224&#39;]]

tb.pe.LPvsB.scaled.breast &lt;- 
  tb.pe.LPvsB.scaled[tb.pe.LPvsB.scaled$GeneEntrezID %in% interestGenes.breast,]

tb.pe.LPvsB.scaled.breast &lt;- 
  tb.pe.LPvsB.scaled.breast[tb.pe.LPvsB.scaled.breast$FDR &lt; 0.05,]

cpm.scaled.filtr.LPvsBasal.breast &lt;- 
  cpm.scaled.filtr.LPvsBasal[tb.pe.LPvsB.scaled.breast$TranscriptName,]

plot.heat.breast &lt;- 
  wrap_elements(grid.grabExpr(draw(foo.heat(cpm.scaled.filtr.LPvsBasal.breast))))


# plotMD does not return invisible(), so we just use wrap_elements
foo.md &lt;- function(x,fontsize = 8){
  par(mar = c(5, 4, 2, 2))
  plotMD(x,xlim = c(-2.5,17.5),
         main = NULL,cex = 0.5,legend = FALSE,
         cex.lab = fontsize/12,
         cex.axis = fontsize/12)
  legend(&#39;topright&#39;,
         legend = c(&#39;NotSig&#39;,&#39;Up&#39;,&#39;Down&#39;),
         pch = rep(16,3), 
         col = c(&#39;black&#39;,&#39;red&#39;,&#39;blue&#39;),
         cex = fontsize/12,
         pt.cex = c(0.3,0.5,0.5))
}
plot.md &lt;- 
  wrap_elements(full = ~foo.md(qlf.pe.LPvsB.scaled))

# plotMDS returns invisible(), we need to manually export the plot (code from limma::plotMDS.MDS)
foo.mds &lt;- function(x,fontsize = 8){
  obj.mds &lt;- plotMDS(x,col = x$samples$Color,main = NULL)
  par(mar = c(5, 4, 2, 2))
  labels &lt;- colnames(obj.mds$distance.matrix.squared)
  StringRadius &lt;- 0.15 * 1 * nchar(labels)
  left.x &lt;- obj.mds$x - StringRadius
  right.x &lt;- obj.mds$x + StringRadius
  Perc &lt;- round(obj.mds$var.explained * 100)
  xlab &lt;- paste(obj.mds$axislabel, 1)
  ylab &lt;- paste(obj.mds$axislabel, 2)
  xlab &lt;- paste0(xlab, &quot; (&quot;, Perc[1], &quot;%)&quot;)
  ylab &lt;- paste0(ylab, &quot; (&quot;, Perc[2], &quot;%)&quot;)
  
  plot(c(-6, 3), c(-3, 3),
       type = &quot;n&quot;,xlab = xlab,ylab = ylab,
       cex.lab = fontsize/12,
       cex.axis = fontsize/12)
  text(obj.mds$x, obj.mds$y, labels = labels, cex = fontsize/12,col = x$samples$Color)
}
plot.mds &lt;- wrap_elements(full = ~foo.mds(dte.pe.scaled.filtr))

plot.design &lt;- c(area(1, 1),area(1,2),area(2, 1),area(2,2))

fig.heatmap &lt;- wrap_plots(A = plot.mds,
                          B = plot.md,
                          C = plot.heat,
                          D = plot.heat.breast,
                          design = plot.design,
                          heights = c(0.4,0.6)) +
  plot_annotation(tag_levels = &#39;a&#39;) 

fig.heatmap &lt;- fig.heatmap &amp;
  theme(plot.tag = element_text(size = 8))

agg_png(filename = file.path(path.misc,&quot;figure6.png&quot;),width = 10,height = 10,units = &#39;in&#39;,res = 300)
fig.heatmap
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<p>Let’s look at the results from the paired-end read data analysis for
the <code>Foxp1-216</code> transcript.</p>
<pre class="r"><code># Foxp1-216
out.pe.LPvsB.scaled$table[&#39;ENSMUST00000175838.2&#39;,]</code></pre>
<pre><code>                     Length EffectiveLength Overdispersion TranscriptName
ENSMUST00000175838.2    420         163.755       1.111807      Foxp1-216
                                    GeneID GeneName GeneEntrezID GeneOfInterest
ENSMUST00000175838.2 ENSMUSG00000030067.18    Foxp1       108655           TRUE
                     NTranscriptPerGene                 Type    logFC
ENSMUST00000175838.2                 31 processed_transcript 1.806999
                         logCPM        F      PValue         FDR
ENSMUST00000175838.2 -0.5914241 14.03927 0.002188547 0.008318195</code></pre>
<p>At the gene-level, we summarize some of the results from the DE
analysis for the <code>Foxp1</code> gene.</p>
<pre class="r"><code>tb.foxp1.gene &lt;- 
  out.pe.LPvsB[grepl(&#39;ENSMUSG00000030067&#39;,out.pe.LPvsB$table$gene_id),]$table
tb.foxp1.gene &lt;- data.frame(tb.foxp1.gene[,-c(2,3,4,5)],row.names = NULL)
setnames(tb.foxp1.gene,
         old = c(&#39;gene_id&#39;,&#39;GeneEntrezID&#39;,&#39;GeneName&#39;,&#39;GeneType&#39;),
         new = c(&#39;Ensembl ID&#39;,&#39;Entrez ID&#39;,&#39;Name&#39;,&#39;Type&#39;))

tb.foxp1.gene$Type &lt;- gsub(&quot;_&quot;,&quot; &quot;,tb.foxp1.gene$Type)

tb.foxp1.gene$logFC &lt;- formatC(round(tb.foxp1.gene$logFC,3),digits = 3,format = &#39;f&#39;)
tb.foxp1.gene$logCPM &lt;- formatC(round(tb.foxp1.gene$logCPM,3),digits = 3,format = &#39;f&#39;)
tb.foxp1.gene$`F` &lt;- formatC(round(tb.foxp1.gene$`F`,3),digits = 3,format = &#39;f&#39;)
tb.foxp1.gene$PValue &lt;- formatC(tb.foxp1.gene$PValue,digits = 3,format = &#39;e&#39;)
tb.foxp1.gene$FDR &lt;- formatC(tb.foxp1.gene$FDR,digits = 3,format = &#39;e&#39;)

tb.foxp1.gene$`Entrez ID` &lt;- NULL
tb.foxp1.gene$Type &lt;- NULL

kb.foxp1.gene &lt;- 
  kbl(tb.foxp1.gene,
      escape = FALSE,
      format = &#39;latex&#39;,
      booktabs = TRUE,
      caption = paste(&#39;edgeR results from a DE analysis at the gene-level for&#39;,
                      &#39;the Foxp1 gene comparing basal and LP cells using the&#39;,
                      &#39;paired-end RNA-seq experiment of the epithelial cell&#39;,
                      &#39;population of the mouse mammary gland&#39;),
      align = &#39;llccccc&#39;) %&gt;%
  kable_styling(font_size = 10)
save_kable(kb.foxp1.gene,file = file.path(path.misc,&quot;supptable_foxp1.tex&quot;))</code></pre>
<p>We can also compare the distribution of the overdispersion estimates
between gene- and transcript-level analyses.</p>
<pre class="r"><code>## Gene-level
df.output.gene &lt;- 
  as.data.table(dge.pe$genes[,c(&#39;GeneName&#39;,&#39;GeneEntrezID&#39;,&#39;GeneType&#39;,&#39;Overdispersion&#39;)])
setnames(df.output.gene,
         old = c(&#39;GeneName&#39;,&#39;GeneEntrezID&#39;,&#39;GeneType&#39;,&#39;Overdispersion&#39;),
         new = c(&#39;Symbol&#39;,&#39;EntrezID&#39;,&#39;Type&#39;,&#39;Overdispersion&#39;))
df.output.gene[,Level := &#39;Gene&#39;]

## Transcript-level
df.output.tx &lt;- data.table(Symbol = catch.pe$annotation$TranscriptName,
                           EntrezID = catch.pe$annotation$GeneEntrezID,
                           Type = catch.pe$annotation$Type,
                           Overdispersion = catch.pe$annotation$Overdispersion,
                           Level = &#39;Transcript&#39;)


df.output.gene.tx &lt;- rbindlist(list(df.output.gene,
                                    df.output.tx))

fig.mao.gene.tx &lt;- ggplot(df.output.gene.tx,aes(x = log10(Overdispersion),y = Level,fill = Level)) +
  geom_boxplot(outlier.alpha = 0.25,fill = &quot;#bebebe&quot;,col = &#39;black&#39;) +
  labs(y = NULL,x = &#39;Mapping ambiguity overdispersion (log10 scale)&#39;) +
  theme_bw(base_size = 8,base_family = &#39;sans&#39;) +
  theme(legend.position = &#39;none&#39;,
        panel.grid = element_blank(),
        axis.text = element_text(colour = &#39;black&#39;,size = 8))

agg_png(filename = file.path(path.misc,&quot;suppfigure_overdispersion.png&quot;),width = 7.5,height = 5,units = &#39;in&#39;,res = 300)
fig.mao.gene.tx
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>fig.mao.gene.tx</code></pre>
<p><img src="figure/mouse.Rmd/mouse_eda_plots_mao_gene-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-mouse_eda_plots_mao_gene-1">
Past versions of mouse_eda_plots_mao_gene-1.png
</button>
</p>
<div id="fig-mouse_eda_plots_mao_gene-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/e75e3d99d906c0c30ba538acfb3516f8fce9547d/docs/figure/mouse.Rmd/mouse_eda_plots_mao_gene-1.png" target="_blank">e75e3d9</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-wf/blob/9a3e9261caa134d79e86c10b792aee7e08f08ff5/docs/figure/mouse.Rmd/mouse_eda_plots_mao_gene-1.png" target="_blank">9a3e926</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Below we have the top 100 genes with largest mapping ambiguity
overdispersion.</p>
<pre class="r"><code>df.output.gene.top &lt;- df.output.gene[order(-Overdispersion),][1:100,]
df.output.gene.top[,Level := NULL]
df.output.gene.top$Type &lt;- 
  gsub(&quot;_&quot;,&quot; &quot;,df.output.gene.top$Type)
df.output.gene.top$EntrezID[is.na(df.output.gene.top$EntrezID)] &lt;- 
  &quot;-&quot;

cap.mao &lt;- paste(&quot;Top 100 genes with largest mapping ambiguity overdispersion.&quot;,
                 &quot;Data from the RNA-seq experiment of the epithelial cell&quot;,
                 &quot;population from the mouse mammary gland generated with&quot;,
                 &quot;paired-end reads.&quot;)

kb.output.gene.top &lt;- kbl(df.output.gene.top,
                          longtable = TRUE,
                          escape = FALSE,
                          format = &#39;latex&#39;,
                          booktabs = TRUE,
                          caption = cap.mao,
                          digits = 2,
                          align = c(&#39;l&#39;,rep(&#39;r&#39;,3)),
                          col.names = c(&#39;Gene Symbol&#39;,&#39;Entrez ID&#39;,&#39;Biotype&#39;,&#39;Overdispersion&#39;)) %&gt;%
  kable_styling(latex_options = c(&quot;scale_down&quot;,&quot;repeat_header&quot;),font_size = 10)</code></pre>
<pre><code>Warning in styling_latex_scale_down(out, table_info): Longtable cannot be
resized.</code></pre>
<pre class="r"><code>save_kable(kb.output.gene.top,file = file.path(path.misc,&quot;supptable_overdispersion.tex&quot;))</code></pre>
</div>
</div>
<div id="analysis-of-single-end-data" class="section level1">
<h1>Analysis of single-end data</h1>
<p>Now, we present a similar analysis but using the single-end read
experiment from the same population of epithelial cells of the mouse
mammay gland.</p>
<div id="transcript-level-analysis-1" class="section level2">
<h2>Transcript-level analysis</h2>
<div id="data-wrangling-1" class="section level3">
<h3>Data wrangling</h3>
<p>We again begin with some data wrangling.</p>
<pre class="r"><code>dt.targets.se &lt;- fread(file.path(path.data.se,&#39;misc/targets.txt&#39;))
dt.targets.se[,Sample := paste(Group,Replicate,sep = &#39;.&#39;)]
dt.targets.se[,Population := mapvalues(Population,&#39;Luminal&#39;,&#39;LP&#39;)]
dt.targets.se$Stage &lt;- strsplit2(dt.targets.se$Group,&quot;\\.&quot;)[,2]
dt.targets.se$Group &lt;- NULL

dt.targets.se[,Color := 
                mapvalues(Population,from = c(&#39;Basal&#39;,&#39;LP&#39;),to = c(&#39;blue&#39;,&#39;darkgreen&#39;))]

dt.targets.se[,path := file.path(path.quant.se,&#39;salmon&#39;,gsub(&#39;.fastq.gz&#39;,&#39;&#39;,File))]</code></pre>
<p>And then we load the output from <code>Salmon</code> into our R
session with the function <code>catchSalmon</code>.</p>
<pre class="r"><code>catch.se &lt;- catchSalmon(dt.targets.se$path,verbose = FALSE)

key.catch.se.anno &lt;- match(rownames(catch.se$annotation),dt.anno$TXIDVERSION)

catch.se$annotation$TranscriptName &lt;- dt.anno$TXEXTERNALNAME[key.catch.se.anno]
catch.se$annotation$GeneID &lt;- dt.anno$GENEIDVERSION[key.catch.se.anno]
catch.se$annotation$GeneName &lt;- dt.anno$GENENAME[key.catch.se.anno]
catch.se$annotation$GeneEntrezID &lt;- dt.anno$ENTREZID[key.catch.se.anno]
catch.se$annotation$GeneOfInterest &lt;- dt.anno$GeneOfInterest[key.catch.se.anno]
catch.se$annotation$NTranscriptPerGene &lt;- dt.anno$NTranscriptPerGene[key.catch.se.anno]</code></pre>
</div>
<div id="differential-transcript-expression-1" class="section level3">
<h3>Differential transcript expression</h3>
<p>In this analysis we only run <code>edgeR</code>. We again compare
scaled and raw counts.</p>
<div id="edger-with-count-scaling-1" class="section level4">
<h4>edgeR with count scaling</h4>
<p>We create the <code>DGEList</code> object using scaled counts.</p>
<pre class="r"><code>dte.se.scaled &lt;- 
  DGEList(counts = catch.se$counts/catch.se$annotation$Overdispersion,
          genes = catch.se$annotation,
          samples = dt.targets.se)
colnames(dte.se.scaled) &lt;- dte.se.scaled$samples$Sample</code></pre>
<p>Perform some filtering:</p>
<pre class="r"><code>keep.se.scaled &lt;- 
  filterByExpr(dte.se.scaled,group = dte.se.scaled$samples$Population) &amp; 
  dte.se.scaled$genes$GeneOfInterest
dte.se.scaled.filtr &lt;- dte.se.scaled[keep.se.scaled,, keep.lib.sizes = FALSE]</code></pre>
<p>Calculate normalization factors, estimate the trended negative
binomial dispersions and perform the QL pipeline:</p>
<pre class="r"><code>design.se &lt;- model.matrix(~Stage + Population,data = dte.se.scaled.filtr$samples)
dte.se.scaled.filtr &lt;- calcNormFactors(dte.se.scaled.filtr)
dte.se.scaled.filtr &lt;- estimateDisp(dte.se.scaled.filtr,design.se,robust = TRUE)</code></pre>
<pre class="r"><code>fit.se.scaled &lt;- glmQLFit(dte.se.scaled.filtr,design.se,robust = TRUE)

qlf.se.LPvsB.scaled &lt;- glmQLFTest(fit.se.scaled)

out.se.LPvsB.scaled &lt;- topTags(qlf.se.LPvsB.scaled,n = Inf)

summary(decideTests(qlf.se.LPvsB.scaled))</code></pre>
<pre><code>       PopulationLP
Down           7895
NotSig         9895
Up             6921</code></pre>
</div>
<div id="edger-with-raw-counts-1" class="section level4">
<h4>edgeR with raw counts</h4>
<p>We repeat the same analysis using raw counts:</p>
<pre class="r"><code>dte.se.raw &lt;- 
  DGEList(counts = catch.se$counts,
          genes = catch.se$annotation,
          samples = dt.targets.se)
colnames(dte.se.raw) &lt;- dte.se.raw$samples$Sample</code></pre>
<pre class="r"><code>keep.se.raw &lt;- 
  filterByExpr(dte.se.raw,group = dte.se.raw$samples$Population) &amp; 
  dte.se.raw$genes$GeneOfInterest
dte.se.raw.filtr &lt;- dte.se.raw[keep.se.raw,,keep.lib.sizes = FALSE]</code></pre>
<pre class="r"><code>dte.se.raw.filtr &lt;- calcNormFactors(dte.se.raw.filtr)
dte.se.raw.filtr &lt;- estimateDisp(dte.se.raw.filtr,design.se,robust = TRUE)</code></pre>
<pre class="r"><code>fit.se.raw &lt;- glmQLFit(dte.se.raw.filtr,design.se,robust = TRUE)

qlf.se.LPvsB.raw &lt;- glmQLFTest(fit.se.raw)

out.se.LPvsB.raw &lt;- topTags(qlf.se.LPvsB.raw,n = Inf)

summary(decideTests(qlf.se.LPvsB.raw))</code></pre>
<pre><code>       PopulationLP
Down           8659
NotSig        21789
Up             6929</code></pre>
</div>
</div>
</div>
<div id="plots-and-other-exploratory-analyses-1" class="section level2">
<h2>Plots and other exploratory analyses</h2>
<p>Below is a MA plot from the DTE results using single-end scaled
data.</p>
<pre class="r"><code>agg_png(filename = file.path(path.misc,&quot;suppfigure_maplot.png&quot;),width = 6,height = 6,units = &#39;in&#39;,res = 300)
plotMD(qlf.se.LPvsB.scaled,xlim = c(-1.5,17.5),
       main = NULL,cex = 0.5,legend = FALSE,
       cex.lab = 8/12,
       cex.axis = 8/12)
legend(&#39;topright&#39;,
       legend = c(&#39;NotSig&#39;,&#39;Up&#39;,&#39;Down&#39;),
       pch = rep(16,3), 
       col = c(&#39;black&#39;,&#39;red&#39;,&#39;blue&#39;),
       cex = 8/12,
       pt.cex = c(0.3,0.5,0.5))
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<p>And we tabulate the results from the analysis using single-end scaled
counts for the Foxp1 gene.</p>
<pre class="r"><code># Foxp1 transcripts (EntrezID 108655)
out.se.LPvsB.scaled$table[out.se.LPvsB.scaled$table$GeneEntrezID %in% &#39;108655&#39;,]</code></pre>
<pre><code>                     Length EffectiveLength Overdispersion TranscriptName
ENSMUST00000177227.8    967             717       1.871821      Foxp1-224
ENSMUST00000177437.8   2468            2218       5.880765      Foxp1-230
ENSMUST00000113322.9   7177            6927      14.765060      Foxp1-204
ENSMUST00000177229.8   1848            1598      30.837225      Foxp1-225
ENSMUST00000177307.8   2121            1871      12.186063      Foxp1-228
                                    GeneID GeneName GeneEntrezID GeneOfInterest
ENSMUST00000177227.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000177437.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000113322.9 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000177229.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000177307.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
                     NTranscriptPerGene     logFC     logCPM         F
ENSMUST00000177227.8                 31 -3.014221  0.6298492 136.33621
ENSMUST00000177437.8                 31 -2.475768  2.5825100  92.30263
ENSMUST00000113322.9                 31 -2.703752  2.5325759  79.71248
ENSMUST00000177229.8                 31 -4.554117 -0.7874704  68.27378
ENSMUST00000177307.8                 31 -1.756318 -0.1121722  13.86188
                           PValue          FDR
ENSMUST00000177227.8 8.014176e-08 1.412541e-06
ENSMUST00000177437.8 6.483437e-07 6.896062e-06
ENSMUST00000113322.9 1.394285e-06 1.269498e-05
ENSMUST00000177229.8 3.085925e-06 2.385992e-05
ENSMUST00000177307.8 3.008508e-03 6.963119e-03</code></pre>
<pre class="r"><code>out.se.LPvsB.raw$table[out.se.LPvsB.raw$table$GeneEntrezID %in% &#39;108655&#39;,]</code></pre>
<pre><code>                     Length EffectiveLength Overdispersion TranscriptName
ENSMUST00000177227.8    967             717       1.871821      Foxp1-224
ENSMUST00000124058.8   1213             963       4.075548      Foxp1-210
ENSMUST00000177437.8   2468            2218       5.880765      Foxp1-230
ENSMUST00000113322.9   7177            6927      14.765060      Foxp1-204
ENSMUST00000177229.8   1848            1598      30.837225      Foxp1-225
ENSMUST00000113326.9   7029            6779     153.195592      Foxp1-206
ENSMUST00000176565.8   3727            3477      23.742224      Foxp1-220
ENSMUST00000177307.8   2121            1871      12.186063      Foxp1-228
                                    GeneID GeneName GeneEntrezID GeneOfInterest
ENSMUST00000177227.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000124058.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000177437.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000113322.9 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000177229.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000113326.9 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000176565.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
ENSMUST00000177307.8 ENSMUSG00000030067.18    Foxp1       108655           TRUE
                     NTranscriptPerGene     logFC     logCPM         F
ENSMUST00000177227.8                 31 -3.213702  0.6403438 81.362703
ENSMUST00000124058.8                 31 -6.522550 -0.1418186 74.459703
ENSMUST00000177437.8                 31 -2.596572  4.2931573 73.376381
ENSMUST00000113322.9                 31 -2.920534  5.5645277 63.534522
ENSMUST00000177229.8                 31 -6.697337  2.9158442 21.558158
ENSMUST00000113326.9                 31 -7.212378  2.1561872 20.217520
ENSMUST00000176565.8                 31 -3.195532  1.2426544 10.122064
ENSMUST00000177307.8                 31 -2.596993  2.4348401  5.705185
                           PValue          FDR
ENSMUST00000177227.8 4.037266e-06 8.099887e-05
ENSMUST00000124058.8 6.043586e-06 1.064018e-04
ENSMUST00000177437.8 7.343061e-06 1.213914e-04
ENSMUST00000113322.9 1.371310e-05 1.853723e-04
ENSMUST00000177229.8 9.674814e-04 4.391203e-03
ENSMUST00000113326.9 1.208199e-03 5.221279e-03
ENSMUST00000176565.8 1.003522e-02 2.790614e-02
ENSMUST00000177307.8 3.852069e-02 8.181077e-02</code></pre>
<pre class="r"><code>tb.se.foxp1.gene &lt;- 
  out.se.LPvsB.scaled[grepl(&#39;ENSMUSG00000030067&#39;,out.se.LPvsB.scaled$table$GeneID),]$table
tb.se.foxp1.gene &lt;- data.frame(&quot;Ensembl ID&quot; = rownames(tb.se.foxp1.gene),
                               tb.se.foxp1.gene[,-c(1,2,3,5,6,7,8,9)],row.names = NULL)

setnames(tb.se.foxp1.gene,
         old = c(&#39;Ensembl.ID&#39;,&#39;TranscriptName&#39;),
         new = c(&#39;Ensembl ID&#39;,&#39;Name&#39;))

tb.se.foxp1.gene$logFC &lt;- formatC(round(tb.se.foxp1.gene$logFC,3),digits = 3,format = &#39;f&#39;)
tb.se.foxp1.gene$logCPM &lt;- formatC(round(tb.se.foxp1.gene$logCPM,3),digits = 3,format = &#39;f&#39;)
tb.se.foxp1.gene$`F` &lt;- formatC(round(tb.se.foxp1.gene$`F`,3),digits = 3,format = &#39;f&#39;)
tb.se.foxp1.gene$PValue &lt;- formatC(tb.se.foxp1.gene$PValue,digits = 3,format = &#39;e&#39;)
tb.se.foxp1.gene$FDR &lt;- formatC(tb.se.foxp1.gene$FDR,digits = 3,format = &#39;e&#39;)

kb.se.foxp1.gene &lt;- 
  kbl(tb.se.foxp1.gene,
      escape = FALSE,
      format = &#39;latex&#39;,
      booktabs = TRUE,
      caption = paste(&quot;edgeR results with count scaling from DTE analysis&quot;,
                      &quot;comparing basal and LP cells for the single-end RNA-seq&quot;,
                      &quot;experiment of the epithelia cell population from the&quot;,
                      &quot;mouse mammary gland. Shown are the topTags results&quot;,
                      &quot;restricted to the Foxp1 transcripts.&quot;),
      align = &quot;llccccc&quot;) %&gt;%
  kable_styling(font_size = 10)
save_kable(kb.se.foxp1.gene,file = file.path(path.misc,&quot;supptable_singleend.tex&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /stornext/System/data/apps/R/R-4.2.1/lib64/R/lib/libRblas.so
LAPACK: /stornext/System/data/apps/R/R-4.2.1/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.22.0            AnnotationFilter_1.22.0    
 [3] GenomicFeatures_1.50.4      AnnotationDbi_1.60.0       
 [5] kableExtra_1.3.4            ragg_1.2.5                 
 [7] SummarizedExperiment_1.28.0 Biobase_2.58.0             
 [9] MatrixGenerics_1.10.0       matrixStats_0.63.0         
[11] forcats_1.0.0               stringr_1.5.0              
[13] dplyr_1.1.0                 purrr_1.0.1                
[15] tidyr_1.3.0                 tidyverse_1.3.2            
[17] tximeta_1.16.1              fishpond_2.4.1             
[19] sleuth_0.30.0               AnnotationHub_3.6.0        
[21] BiocFileCache_2.6.0         dbplyr_2.3.0               
[23] tidyHeatmap_1.8.1           tibble_3.1.8               
[25] patchwork_1.1.2             ComplexHeatmap_2.14.0      
[27] gplots_3.1.3                plyr_1.8.8                 
[29] ggpubr_0.6.0                magrittr_2.0.3             
[31] rtracklayer_1.58.0          GenomicRanges_1.50.2       
[33] GenomeInfoDb_1.34.9         IRanges_2.32.0             
[35] S4Vectors_0.36.1            BiocGenerics_0.44.0        
[37] Rsubread_2.12.2             readr_2.1.4                
[39] ggplot2_3.4.1               data.table_1.14.6          
[41] edgeR_3.40.2                limma_3.54.1               
[43] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.3                    tidyselect_1.2.0             
  [3] RSQLite_2.2.20                BiocParallel_1.32.5          
  [5] munsell_0.5.0                 codetools_0.2-19             
  [7] statmod_1.5.0                 withr_2.5.0                  
  [9] colorspace_2.1-0              filelock_1.0.2               
 [11] highr_0.10                    knitr_1.42                   
 [13] rstudioapi_0.14               SingleCellExperiment_1.20.0  
 [15] ggsignif_0.6.4                labeling_0.4.2               
 [17] git2r_0.31.0                  tximport_1.26.1              
 [19] GenomeInfoDbData_1.2.9        farver_2.1.1                 
 [21] bit64_4.0.5                   rhdf5_2.42.0                 
 [23] rprojroot_2.0.3               vctrs_0.5.2                  
 [25] generics_0.1.3                xfun_0.37                    
 [27] timechange_0.2.0              R6_2.5.1                     
 [29] doParallel_1.0.17             clue_0.3-64                  
 [31] locfit_1.5-9.7                gridGraphics_0.5-1           
 [33] bitops_1.0-7                  rhdf5filters_1.10.0          
 [35] cachem_1.0.6                  DelayedArray_0.24.0          
 [37] assertthat_0.2.1              vroom_1.6.1                  
 [39] promises_1.2.0.1              BiocIO_1.8.0                 
 [41] scales_1.2.1                  googlesheets4_1.0.1          
 [43] gtable_0.3.1                  Cairo_1.6-0                  
 [45] processx_3.8.0                rlang_1.0.6                  
 [47] systemfonts_1.0.4             splines_4.2.1                
 [49] GlobalOptions_0.1.2           rstatix_0.7.2                
 [51] lazyeval_0.2.2                gargle_1.3.0                 
 [53] broom_1.0.3                   reshape2_1.4.4               
 [55] modelr_0.1.10                 BiocManager_1.30.19          
 [57] yaml_2.3.7                    abind_1.4-5                  
 [59] backports_1.4.1               httpuv_1.6.5                 
 [61] qvalue_2.30.0                 tools_4.2.1                  
 [63] ellipsis_0.3.2                jquerylib_0.1.4              
 [65] RColorBrewer_1.1-3            Rcpp_1.0.10                  
 [67] progress_1.2.2                zlibbioc_1.44.0              
 [69] RCurl_1.98-1.10               ps_1.7.2                     
 [71] prettyunits_1.1.1             GetoptLong_1.0.5             
 [73] viridis_0.6.2                 haven_2.5.1                  
 [75] cluster_2.1.4                 fs_1.6.1                     
 [77] svMisc_1.2.3                  circlize_0.4.15              
 [79] reprex_2.0.2                  googledrive_2.0.0            
 [81] whisker_0.4.1                 ProtGenerics_1.30.0          
 [83] hms_1.1.2                     mime_0.12                    
 [85] evaluate_0.20                 xtable_1.8-4                 
 [87] XML_3.99-0.13                 readxl_1.4.2                 
 [89] gridExtra_2.3                 shape_1.4.6                  
 [91] compiler_4.2.1                biomaRt_2.54.0               
 [93] KernSmooth_2.23-20            crayon_1.5.2                 
 [95] htmltools_0.5.4               mgcv_1.8-41                  
 [97] later_1.3.0                   tzdb_0.3.0                   
 [99] lubridate_1.9.2               DBI_1.1.3                    
[101] rappdirs_0.3.3                Matrix_1.5-3                 
[103] car_3.1-1                     cli_3.6.0                    
[105] parallel_4.2.1                pkgconfig_2.0.3              
[107] getPass_0.2-2                 GenomicAlignments_1.34.0     
[109] xml2_1.3.3                    foreach_1.5.2                
[111] svglite_2.1.1                 bslib_0.4.2                  
[113] webshot_0.5.4                 XVector_0.38.0               
[115] rvest_1.0.3                   callr_3.7.3                  
[117] digest_0.6.31                 Biostrings_2.66.0            
[119] cellranger_1.1.0              rmarkdown_2.20               
[121] dendextend_1.16.0             restfulr_0.0.15              
[123] curl_5.0.0                    shiny_1.7.4                  
[125] Rsamtools_2.14.0              gtools_3.9.4                 
[127] rjson_0.2.21                  nlme_3.1-162                 
[129] lifecycle_1.0.3               jsonlite_1.8.4               
[131] Rhdf5lib_1.20.0               carData_3.0-5                
[133] viridisLite_0.4.1             fansi_1.0.4                  
[135] pillar_1.8.1                  lattice_0.20-45              
[137] KEGGREST_1.38.0               fastmap_1.1.0                
[139] httr_1.4.4                    interactiveDisplayBase_1.36.0
[141] glue_1.6.2                    png_0.1-8                    
[143] iterators_1.0.14              BiocVersion_3.16.0           
[145] bit_4.0.5                     stringi_1.7.12               
[147] sass_0.4.1                    blob_1.2.3                   
[149] textshaping_0.3.6             caTools_1.18.2               
[151] memoise_2.0.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
