<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Pedro L. Baldoni" />


<title>Simulation - Reduced report (results presented in the main paper)</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TranscriptDE-code</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/plbaldoni/TranscriptDE-code">
    <span class="fab fa-github"></span>
     
    GitHub
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulation - Reduced report (results
presented in the main paper)</h1>
<h4 class="author">Pedro L. Baldoni</h4>
<h4 class="date">01 August, 2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>TranscriptDE-code/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221115code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221115)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221115code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221115)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomplbaldoniTranscriptDEcodetree38d0392d56cd4471c48e8371aea51467acb61489targetblank38d0392a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/plbaldoni/TranscriptDE-code/tree/38d0392d56cd4471c48e8371aea51467acb61489" target="_blank">38d0392</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomplbaldoniTranscriptDEcodetree38d0392d56cd4471c48e8371aea51467acb61489targetblank38d0392a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/plbaldoni/TranscriptDE-code/tree/38d0392d56cd4471c48e8371aea51467acb61489" target="_blank">38d0392</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    .gitignore
    Ignored:    TranscriptDE-code.Rproj
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    code/lung-se/data/slurm-10685114.out
    Ignored:    code/lung-se/salmon/.RData
    Ignored:    code/lung-se/salmon/runWasabi.Rout
    Ignored:    code/lung-se/salmon/slurm-10685171.out
    Ignored:    code/lung-se/salmon/slurm-10694099.out
    Ignored:    code/lung/data/slurm-10678225.out
    Ignored:    code/lung/index/slurm-10679764.out
    Ignored:    code/lung/index/slurm-10679768.out
    Ignored:    code/lung/index/slurm-10684814.out
    Ignored:    code/lung/salmon/.RData
    Ignored:    code/lung/salmon/runWasabi.Rout
    Ignored:    code/lung/salmon/slurm-10681840.out
    Ignored:    code/lung/salmon/slurm-10681872.out
    Ignored:    code/lung/salmon/slurm-10684950.out
    Ignored:    code/lung/salmon/slurm-10694066.out
    Ignored:    code/pkg/.Rhistory
    Ignored:    code/pkg/.Rproj.user/
    Ignored:    code/pkg/pkg.Rproj
    Ignored:    code/pkg/src/RcppExports.o
    Ignored:    code/pkg/src/pkg.so
    Ignored:    code/pkg/src/rcpparma_hello_world.o
    Ignored:    data/annotation/hg38/
    Ignored:    data/annotation/mm39/
    Ignored:    data/annotation/sequins/._rnasequin_annotation_2.4.gtf
    Ignored:    data/annotation/sequins/._rnasequin_decoychr_2.4.fa
    Ignored:    data/annotation/sequins/._rnasequin_decoychr_2.4.fa.fai
    Ignored:    data/annotation/sequins/._rnasequin_genes_2.4.tsv
    Ignored:    data/annotation/sequins/._rnasequin_isoforms_2.4.tsv
    Ignored:    data/annotation/sequins/._rnasequin_sequences_2.4.fa
    Ignored:    data/lung-se/fastq/
    Ignored:    data/lung-se/misc/._filereport_read_run_PRJNA341465_tsv.txt
    Ignored:    data/lung/fastq/
    Ignored:    data/lung/index/
    Ignored:    data/lung/misc/._filereport_read_run_PRJNA723287_tsv.txt
    Ignored:    ignore/
    Ignored:    misc/.DS_Store
    Ignored:    misc/._.DS_Store
    Ignored:    misc/casestudy.Rmd/._figure6.png
    Ignored:    misc/casestudy.Rmd/._suppfigure_maplot.png
    Ignored:    misc/casestudy.Rmd/._suppfigure_overdispersion.png
    Ignored:    misc/casestudy.Rmd/._suppfigure_venn.png
    Ignored:    misc/casestudy.Rmd/._supptable_gene.tex
    Ignored:    misc/casestudy.Rmd/._supptable_overdispersion.tex
    Ignored:    misc/simulation-paper.Rmd/._figure2.png
    Ignored:    misc/simulation-paper.Rmd/._figure5.png
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/lung-se/
    Ignored:    output/lung/
    Ignored:    output/quasi_poisson/
    Ignored:    output/simulation/

Untracked files:
    Untracked:  misc/casestudy.Rmd/figure2.pdf
    Untracked:  misc/casestudy.Rmd/figure7.pdf
    Untracked:  misc/simulation-paper.Rmd/figure6.pdf

Unstaged changes:
    Modified:   misc/casestudy.Rmd/figure1.pdf
    Deleted:    misc/casestudy.Rmd/figure6.pdf
    Deleted:    misc/simulation-paper.Rmd/figure2.pdf
    Modified:   misc/simulation-paper.Rmd/figure3.pdf
    Modified:   misc/simulation-paper.Rmd/figure4.pdf
    Modified:   misc/simulation-paper.Rmd/figure5.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/simulation-paper.Rmd</code>) and
HTML (<code>docs/simulation-paper.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/38d0392d56cd4471c48e8371aea51467acb61489/analysis/simulation-paper.Rmd" target="_blank">38d0392</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-07-31
</td>
<td>
Adding BCV plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/3e118096971a7aeabae21ea5c4ddd7b21a8bda2f/docs/simulation-paper.html" target="_blank">3e11809</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-06-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/6a742643fec485d8f2bc5252a13a61c6b6edf2e5/analysis/simulation-paper.Rmd" target="_blank">6a74264</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-06-05
</td>
<td>
Producing figures in PDF format for the main paper
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/6a742643fec485d8f2bc5252a13a61c6b6edf2e5/docs/simulation-paper.html" target="_blank">6a74264</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-06-05
</td>
<td>
Producing figures in PDF format for the main paper
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/bfd4304523dda9223897c987496afecaef34b5ad/docs/simulation-paper.html" target="_blank">bfd4304</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-05-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/1a75b185cb0bf29fc175bf1eaece7f559e2e96aa/analysis/simulation-paper.Rmd" target="_blank">1a75b18</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-05-21
</td>
<td>
updating results of single-end experiments
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/5610c90eb61e6128c7c2e1bff332046b7718241d/docs/simulation-paper.html" target="_blank">5610c90</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/f01c7b4c2b01f0cdbe011f33e7e961e756fa140d/docs/simulation-paper.html" target="_blank">f01c7b4</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/e2ef9d67c66f54984d089ef512df023e4056eb46/analysis/simulation-paper.Rmd" target="_blank">e2ef9d6</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-23
</td>
<td>
Removing cache
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/24d66f4fd8d7636071951744185d01574800fcd4/docs/simulation-paper.html" target="_blank">24d66f4</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/9b79374f238a8ca13ceddb73ae75b50f5e033720/analysis/simulation-paper.Rmd" target="_blank">9b79374</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Fix typo with sans family
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/6a33f36051b3cc3197e6702589169bd914f98d04/docs/simulation-paper.html" target="_blank">6a33f36</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/d42adad7cb9d9aa8b34e71e381daaf67f25c6766/analysis/simulation-paper.Rmd" target="_blank">d42adad</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Presenting histograms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/38286e354e42d2e5148adbbc9e5a1f7d05bd2ca8/docs/simulation-paper.html" target="_blank">38286e3</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/b8e397902f76d2cf942e92fb1eb38fffc7ef1eb3/docs/simulation-paper.html" target="_blank">b8e3979</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/57b0d0081f0113c473a272fb1b8ef921d68ee788/analysis/simulation-paper.Rmd" target="_blank">57b0d00</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Renaming repo and organizing main page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/57b0d0081f0113c473a272fb1b8ef921d68ee788/docs/simulation-paper.html" target="_blank">57b0d00</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
<td>
Renaming repo and organizing main page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/623d4297221b6b8a0e0e44292e19626bf6ed3270/analysis/simulation-paper.Rmd" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Splitting figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/623d4297221b6b8a0e0e44292e19626bf6ed3270/docs/simulation-paper.html" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
<td>
Splitting figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/49c9a940256334199278c62ffc22b49d22fa6883/analysis/simulation-paper.Rmd" target="_blank">49c9a94</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-19
</td>
<td>
Expanding panels to multiple figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/49c9a940256334199278c62ffc22b49d22fa6883/docs/simulation-paper.html" target="_blank">49c9a94</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-19
</td>
<td>
Expanding panels to multiple figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/4276bfc1b4f2a31caae390bb910515ab77e165cf/analysis/simulation-paper.Rmd" target="_blank">4276bfc</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-06
</td>
<td>
Organizing output of latex table
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/4276bfc1b4f2a31caae390bb910515ab77e165cf/docs/simulation-paper.html" target="_blank">4276bfc</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-06
</td>
<td>
Organizing output of latex table
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/a8c51afeef4ad8b3b3c30c8bbae530844427d5af/analysis/simulation-paper.Rmd" target="_blank">a8c51af</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-05
</td>
<td>
Updating simulation-paper report
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/a8c51afeef4ad8b3b3c30c8bbae530844427d5af/docs/simulation-paper.html" target="_blank">a8c51af</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-05
</td>
<td>
Updating simulation-paper report
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/d34d4e689994129c6bbdc8a356fe5778afbdd1e1/analysis/simulation-paper.Rmd" target="_blank">d34d4e6</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-24
</td>
<td>
Adding simulation-paper-report
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/plbaldoni/TranscriptDE-code/d34d4e689994129c6bbdc8a356fe5778afbdd1e1/docs/simulation-paper.html" target="_blank">d34d4e6</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-24
</td>
<td>
Adding simulation-paper-report
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this report, we present the analysis of the simulations for the
<code>catchSalmon</code>/<code>catchKallisto</code> manuscript. These
simulations aim to generate typical RNA-seq data from mouse experiments.
This report focuses on the results presented in the main paper only. For
a comprehensive report of the results, please refer to the <a
href="simulation-complete.html">complete report</a>.</p>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>We load necessary libraries and set up the rendering options
below.</p>
<pre class="r"><code>knitr::opts_chunk$set(dev = &quot;png&quot;,
                      dpi = 300,
                      dev.args = list(type = &quot;cairo-png&quot;),
                      root.dir = &#39;.&#39;,
                      autodep = TRUE)

options(knitr.kable.NA = &quot;-&quot;)</code></pre>
<pre class="r"><code>library(data.table)
library(ggplot2)
library(thematic)
library(plyr)
library(magrittr)
library(limma)
library(edgeR)
library(BiocParallel)
library(devtools)
library(purrr)
library(readr)
library(ggpubr)
library(kableExtra)
library(patchwork)
library(ragg)
load_all(&#39;../code/pkg/&#39;)</code></pre>
<p>I use the functions below to produce the histogram plot shown in this
report and to quickly subset data tables for specific scenarios.</p>
<pre class="r"><code>cleanPlot &lt;- function(x,fig){
  if (x == max(seq_along(fig))) {
    y &lt;- fig[[x]]
  } else{
    y &lt;- fig[[x]] + theme(axis.title.x = element_blank(),
                          axis.text.x = element_blank(),
                          axis.ticks.x = element_blank())
  }
  if (x &gt; 1) {
    y &lt;- y + theme(strip.background.x = element_blank(),
                   strip.text.x = element_blank())
  }
  return(y)
}

subsetDT &lt;- function(x,scenario,panel = NULL,tx.per.gene = NULL, plot = TRUE){
  if(isTRUE(plot)){
    if(panel %in% c(&#39;A&#39;,&#39;B&#39;)){
      out &lt;- x[Genome == scenario[&#39;genome&#39;] &amp;
                 FC == ifelse(panel == &#39;A&#39;,&#39;fc2&#39;,&#39;fc1&#39;) &amp; 
                 Length == scenario[&#39;length&#39;] &amp;
                 Reads == scenario[&#39;read&#39;] &amp; 
                 Quantifier == scenario[&#39;quantifier&#39;] &amp; 
                 Scenario == scenario[&#39;scenario&#39;],]
    } else{
      out &lt;- x[Genome == scenario[&#39;genome&#39;] &amp;
                 FC == &#39;fc1&#39; &amp; 
                 Length == scenario[&#39;length&#39;] &amp;
                 Reads == scenario[&#39;read&#39;] &amp; 
                 Quantifier == scenario[&#39;quantifier&#39;] &amp; 
                 Scenario == scenario[&#39;scenario&#39;] &amp;
                 TxPerGene == tx.per.gene ,]
    }
  } else{
    out &lt;- x[Genome == scenario[&#39;genome&#39;] &amp;
               FC == &#39;fc2&#39; &amp; 
               Quantifier == scenario[&#39;quantifier&#39;] &amp; 
               TxPerGene == scenario[&#39;txpergene&#39;],]
  }
  return(out)
}</code></pre>
</div>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<p>Here we begin summarizing the results to generate the figures
presented in the main paper.</p>
<div id="data-wrangling" class="section level2">
<h2>Data wrangling</h2>
<p>Below I set up the file paths.</p>
<pre class="r"><code>path.misc &lt;- file.path(&#39;../misc&#39;,knitr::current_input())
dir.create(path.misc,recursive = TRUE,showWarnings = FALSE)

path.fdr &lt;- 
  list.files(&#39;../output/simulation/summary&#39;,&#39;fdr.tsv.gz&#39;,recursive = TRUE,full.names = TRUE)
path.metrics &lt;- 
  list.files(&#39;../output/simulation/summary&#39;,&#39;metrics.tsv.gz&#39;,recursive = TRUE,full.names = TRUE)
path.time &lt;- 
  list.files(&#39;../output/simulation/summary&#39;,&#39;time.tsv.gz&#39;,recursive = TRUE,full.names = TRUE)
path.quantile &lt;- 
  list.files(&#39;../output/simulation/summary&#39;,&#39;quantile.tsv.gz&#39;,recursive = TRUE,full.names = TRUE)
path.pvalue &lt;- 
  list.files(&#39;../output/simulation/summary&#39;,&#39;pvalue.tsv.gz&#39;,recursive = TRUE,full.names = TRUE)
path.overdispersion &lt;- 
  list.files(&#39;../output/simulation/summary&#39;,&#39;overdispersion.tsv.gz&#39;,recursive = TRUE,full.names = TRUE)</code></pre>
<p>Loading all summarized results below.</p>
<pre class="r"><code># Loading datasets
dt.fdr &lt;- do.call(rbind,lapply(path.fdr,fread))
dt.metrics &lt;- do.call(rbind,lapply(path.metrics,fread))
dt.time &lt;- do.call(rbind,lapply(path.time,fread))
dt.quantile &lt;- do.call(rbind,lapply(path.quantile,fread))
dt.pvalue &lt;- do.call(rbind,lapply(path.pvalue,fread))
dt.overdispersion &lt;- do.call(rbind,lapply(path.overdispersion,fread))</code></pre>
<p>Some data wrangling below.</p>
<pre class="r"><code># Changing labels
dt.fdr$TxPerGene %&lt;&gt;%
  mapvalues(from = paste0(c(2, 3, 4, 5, 9999), &#39;TxPerGene&#39;),
            to = c(paste0(&quot;#Tx/Gene = &quot;, c(2, 3, 4, 5)), &#39;All Transcripts&#39;))
dt.fdr$LibsPerGroup %&lt;&gt;%
  mapvalues(from = paste0(c(3, 5, 10), &#39;libsPerGroup&#39;),
            to = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10))) %&gt;%
  factor(levels = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10)))
dt.fdr$Quantifier %&lt;&gt;% mapvalues(from = &#39;salmon&#39;, to = &#39;Salmon&#39;)
dt.fdr$Length %&lt;&gt;% mapvalues(from = paste0(&#39;readlen-&#39;, seq(50, 150, 25)),
                             to = paste0(seq(50, 150, 25), &#39;bp&#39;))

dt.metrics$TxPerGene %&lt;&gt;%
  mapvalues(from = paste0(c(2, 3, 4, 5, 9999), &#39;TxPerGene&#39;),
            to = c(paste0(&quot;#Tx/Gene = &quot;, c(2, 3, 4, 5)), &#39;All Transcripts&#39;))
dt.metrics$LibsPerGroup %&lt;&gt;%
  mapvalues(from = paste0(c(3, 5, 10), &#39;libsPerGroup&#39;),
            to = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10))) %&gt;%
  factor(levels = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10)))
dt.metrics$Quantifier %&lt;&gt;% mapvalues(from = &#39;salmon&#39;, to = &#39;Salmon&#39;)
dt.metrics$Length %&lt;&gt;% mapvalues(from = paste0(&#39;readlen-&#39;, seq(50, 150, 25)),
                                 to = paste0(seq(50, 150, 25), &#39;bp&#39;))

dt.time$TxPerGene %&lt;&gt;%
  mapvalues(from = paste0(c(2, 3, 4, 5, 9999), &#39;TxPerGene&#39;),
            to = c(paste0(&quot;#Tx/Gene = &quot;, c(2, 3, 4, 5)), &#39;All Transcripts&#39;))
dt.time$LibsPerGroup %&lt;&gt;%
  mapvalues(from = paste0(c(3, 5, 10), &#39;libsPerGroup&#39;),
            to = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10))) %&gt;%
  factor(levels = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10)))
dt.time$Quantifier %&lt;&gt;% mapvalues(from = &#39;salmon&#39;, to = &#39;Salmon&#39;)
dt.time$Length %&lt;&gt;% mapvalues(from = paste0(&#39;readlen-&#39;, seq(50, 150, 25)),
                              to = paste0(seq(50, 150, 25), &#39;bp&#39;))

dt.quantile$TxPerGene %&lt;&gt;%
  mapvalues(from = paste0(c(2, 3, 4, 5, 9999), &#39;TxPerGene&#39;),
            to = c(paste0(&quot;#Tx/Gene = &quot;, c(2, 3, 4, 5)), &#39;All Transcripts&#39;))
dt.quantile$LibsPerGroup %&lt;&gt;%
  mapvalues(from = paste0(c(3, 5, 10), &#39;libsPerGroup&#39;),
            to = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10))) %&gt;%
  factor(levels = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10)))
dt.quantile$Quantifier %&lt;&gt;% mapvalues(from = &#39;salmon&#39;, to = &#39;Salmon&#39;)
dt.quantile$Length %&lt;&gt;% mapvalues(from = paste0(&#39;readlen-&#39;, seq(50, 150, 25)),
                                  to = paste0(seq(50, 150, 25), &#39;bp&#39;))

dt.pvalue$TxPerGene %&lt;&gt;%
  mapvalues(from = paste0(c(2, 3, 4, 5, 9999), &#39;TxPerGene&#39;),
            to = c(paste0(&quot;#Tx/Gene = &quot;, c(2, 3, 4, 5)), &#39;All Transcripts&#39;))
dt.pvalue$LibsPerGroup %&lt;&gt;%
  mapvalues(from = paste0(c(3, 5, 10), &#39;libsPerGroup&#39;),
            to = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10))) %&gt;%
  factor(levels = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10)))
dt.pvalue$Quantifier %&lt;&gt;% mapvalues(from = &#39;salmon&#39;, to = &#39;Salmon&#39;)
dt.pvalue$Length %&lt;&gt;% mapvalues(from = paste0(&#39;readlen-&#39;, seq(50, 150, 25)),
                                to = paste0(seq(50, 150, 25), &#39;bp&#39;))

dt.overdispersion$TxPerGene %&lt;&gt;%
  mapvalues(from = paste0(c(2, 3, 4, 5, 9999), &#39;TxPerGene&#39;),
            to = c(paste0(&quot;#Tx/Gene = &quot;, c(2, 3, 4, 5)), &#39;All Transcripts&#39;))
dt.overdispersion$LibsPerGroup %&lt;&gt;%
  mapvalues(from = paste0(c(3, 5, 10), &#39;libsPerGroup&#39;),
            to = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10))) %&gt;%
  factor(levels = paste0(&#39;#Lib/Group = &#39;, c(3, 5, 10)))
dt.overdispersion$Quantifier %&lt;&gt;% mapvalues(from = &#39;salmon&#39;, to = &#39;Salmon&#39;)
dt.overdispersion$Length %&lt;&gt;% mapvalues(from = paste0(&#39;readlen-&#39;, seq(50, 150, 25)),
                                        to = paste0(seq(50, 150, 25), &#39;bp&#39;))</code></pre>
<p>All the simulated scenarios are generated below.</p>
<pre class="r"><code>dt.scenario &lt;- expand.grid(&#39;genome&#39; = &#39;mm39&#39;,
                           &#39;length&#39; = c(&#39;50bp&#39;,&#39;75bp&#39;,&#39;100bp&#39;,&#39;125bp&#39;,&#39;150bp&#39;),
                           &#39;read&#39; = c(&#39;single-end&#39;,&#39;paired-end&#39;),
                           &#39;quantifier&#39; = c(&#39;Salmon&#39;,&#39;kallisto&#39;),
                           &#39;scenario&#39; = c(&#39;balanced&#39;,&#39;unbalanced&#39;),
                           stringsAsFactors = FALSE)
dt.scenario &lt;- as.data.table(dt.scenario)</code></pre>
</div>
<div id="power-plot-false-discovery-rate" class="section level2">
<h2>Power plot &amp; False discovery rate</h2>
<p>Below we generate Figure 3 of the main paper.</p>
<pre class="r"><code>scenario.balanced &lt;- as.character(dt.scenario[length == &#39;100bp&#39; &amp;
                                                read == &#39;paired-end&#39; &amp;
                                                quantifier == &#39;Salmon&#39; &amp;
                                                scenario == &#39;balanced&#39;,])
scenario.unbalanced &lt;- as.character(dt.scenario[length == &#39;100bp&#39; &amp;
                                                  read == &#39;paired-end&#39; &amp;
                                                  quantifier == &#39;Salmon&#39; &amp;
                                                  scenario == &#39;unbalanced&#39;,])
names(scenario.balanced) &lt;- colnames(dt.scenario)
names(scenario.unbalanced) &lt;- colnames(dt.scenario)


dt.power &lt;- rbind(subsetDT(dt.metrics,scenario.balanced,&#39;A&#39;),
                  subsetDT(dt.metrics,scenario.unbalanced,&#39;A&#39;))

dt.power &lt;- dt.power[LibsPerGroup != &#39;#Lib/Group = 10&#39;,]

dt.power$LibsPerGroup %&lt;&gt;% mapvalues(from = paste0(&#39;#Lib/Group = &#39;, c(3, 5)),
                                     to = paste0(c(3,5),&#39; samples per group&#39;))

dt.power$Scenario %&lt;&gt;% 
  mapvalues(from = c(&#39;balanced&#39;,&#39;unbalanced&#39;),
            to = c(&#39;Equal library sizes&#39;,&#39;Unequal library sizes&#39;))

dt.power[, FDR := roundPretty(ifelse((FP+TP) == 0,NA,100*FP/(FP+TP)),1)]

dt.power &lt;- dt.power[TxPerGene == &#39;All Transcripts&#39;,]

sub.byvar &lt;- 
  colnames(dt.power)[-which(colnames(dt.power) %in% c(&#39;P.SIG&#39;,&#39;TP&#39;,&#39;FP&#39;))]

gap &lt;- 0.05*max(dt.power$TP + dt.power$FP)

x.melt &lt;- melt(dt.power,id.vars = sub.byvar,
               measure.vars = c(&#39;TP&#39;,&#39;FP&#39;),
               variable.name = &#39;Type&#39;,
               value.name = &#39;Value&#39;)
x.melt$Type &lt;- 
  factor(x.melt$Type,
         levels = c(&#39;FP&#39;,&#39;TP&#39;),
         labels = c(&#39;False&#39;,&#39;True&#39;))

plot.power &lt;- function(df.bar,df.txt,scenario,library,legend = FALSE, base_size = 8){
  tb.bar &lt;- df.bar[Scenario == scenario &amp; LibsPerGroup == library,]
  tb.txt &lt;- df.txt[Scenario == scenario &amp; LibsPerGroup == library,][FDR != &#39;NA&#39;,]
  
  ggplot(tb.bar,aes(x = Method,y = Value,fill = Type)) +
    geom_col(colour = &#39;black&#39;) +
    geom_text(aes(x = Method,y = (TP + FP) + gap,label = FDR),
              vjust = 0,data = tb.txt,size = base_size/.pt,inherit.aes = FALSE) +
    scale_fill_manual(values = c(&#39;#ff0000&#39;,&#39;#bebebe&#39;)) +
    labs(x = NULL,y = paste(&#39;DE Transcripts&#39;)) +
    scale_y_continuous(limits = c(0,3000)) +
    theme_bw(base_size = base_size,base_family = &#39;sans&#39;) +
    theme(panel.grid = element_blank(),
          axis.text.x = element_text(angle = 90),
          axis.text = element_text(colour = &#39;black&#39;,size = base_size)) +
    if (legend == TRUE) theme(legend.background = element_rect(fill = alpha(&#39;white&#39;, 0)),
                              legend.text = element_text(size = base_size),
                              legend.position = c(0.80,0.90),legend.title = element_blank(),
                              legend.key.size = unit(0.75,&quot;line&quot;)) else theme(legend.position = &#39;none&#39;)
}

fig.power.a &lt;- plot.power(df.bar = x.melt,df.txt = dt.power,scenario = &#39;Equal library sizes&#39;,library = &#39;3 samples per group&#39;)
fig.power.b &lt;- plot.power(df.bar = x.melt,df.txt = dt.power,scenario = &#39;Unequal library sizes&#39;,library = &#39;3 samples per group&#39;,legend = TRUE)
fig.power.c &lt;- plot.power(df.bar = x.melt,df.txt = dt.power,scenario = &#39;Equal library sizes&#39;,library = &#39;5 samples per group&#39;)
fig.power.d &lt;- plot.power(df.bar = x.melt,df.txt = dt.power,scenario = &#39;Unequal library sizes&#39;,library = &#39;5 samples per group&#39;)

fig.power &lt;- (fig.power.a + fig.power.b) / (fig.power.c + fig.power.d) +
  plot_annotation(tag_levels = &#39;a&#39;) +
  theme(plot.tag = element_text(size = 8))

ggsave(plot = fig.power,
       filename = file.path(path.misc,&quot;figure3.pdf&quot;),
       device = &#39;pdf&#39;,width = 5,height = 5,units = &#39;in&#39;,dpi = 300)

fig.power</code></pre>
<p><img src="figure/simulation-paper.Rmd/simulation-paper_power-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-simulation-paper_power-1">
Past versions of simulation-paper_power-1.png
</button>
</p>
<div id="fig-simulation-paper_power-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/623d4297221b6b8a0e0e44292e19626bf6ed3270/docs/figure/simulation-paper.Rmd/simulation-paper_power-1.png" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/49c9a940256334199278c62ffc22b49d22fa6883/docs/figure/simulation-paper.Rmd/simulation-paper_power-1.png" target="_blank">49c9a94</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/5dcd60b1e32ad8f39163c4ac3727c17220cb5ad1/docs/figure/simulation-paper.Rmd/simulation-paper_power-1.png" target="_blank">5dcd60b</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/d34d4e689994129c6bbdc8a356fe5778afbdd1e1/docs/figure/simulation-paper.Rmd/simulation-paper_power-1.png" target="_blank">d34d4e6</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Then, we generate Figure 4 below.</p>
<pre class="r"><code>dt.fdr.plot &lt;- rbind(subsetDT(dt.fdr,scenario.balanced,&#39;A&#39;),
                     subsetDT(dt.fdr,scenario.unbalanced,&#39;A&#39;))

dt.fdr.plot &lt;- dt.fdr.plot[LibsPerGroup != &quot;#Lib/Group = 10&quot;,]

dt.fdr.plot$LibsPerGroup %&lt;&gt;% 
  mapvalues(from = paste0(&#39;#Lib/Group = &#39;, c(3, 5)),
            to = paste0(c(3,5),&#39; samples per group&#39;))

dt.fdr.plot$Scenario %&lt;&gt;% 
  mapvalues(from = c(&#39;balanced&#39;,&#39;unbalanced&#39;),
            to = c(&#39;Equal library sizes&#39;,&#39;Unequal library sizes&#39;))

dt.fdr.plot &lt;- dt.fdr.plot[TxPerGene == &#39;All Transcripts&#39;,]

plot.fdr &lt;- function(df.line,scenario,library,legend = FALSE,base_size = 8){
  tb.bar &lt;- df.line[Scenario == scenario &amp; LibsPerGroup == library,]
  
  ggplot(tb.bar,aes(x = N,y = FDR,color = Method,group = Method)) +
    geom_line(linewidth = 0.5) +
    scale_color_manual(values = methodsNames()$color) +
    scale_y_continuous(limits = c(0,1250)) +
    labs(y = &#39;False discoveries&#39;,x = &#39;Transcripts chosen&#39;) +
    theme_bw(base_size = base_size,base_family = &#39;sans&#39;) +
    theme(panel.grid = element_blank(),
          axis.text = element_text(colour = &#39;black&#39;,size = base_size)) +
    if (legend == TRUE) theme(legend.background = element_rect(fill = alpha(&#39;white&#39;, 0)),
                              legend.direction = &#39;vertical&#39;,
                              legend.position = c(0.3,0.8),
                              legend.text = element_text(size = base_size),
                              legend.title = element_blank(),
                              legend.key.size = unit(0.75,&quot;line&quot;)) else theme(legend.position = &#39;none&#39;)
}

fig.fdr.a &lt;- plot.fdr(df.line = dt.fdr.plot,scenario = &#39;Equal library sizes&#39;,library = &#39;3 samples per group&#39;)
fig.fdr.b &lt;- plot.fdr(df.line = dt.fdr.plot,scenario = &#39;Unequal library sizes&#39;,library = &#39;3 samples per group&#39;,legend = TRUE)
fig.fdr.c &lt;- plot.fdr(df.line = dt.fdr.plot,scenario = &#39;Equal library sizes&#39;,library = &#39;5 samples per group&#39;)
fig.fdr.d &lt;- plot.fdr(df.line = dt.fdr.plot,scenario = &#39;Unequal library sizes&#39;,library = &#39;5 samples per group&#39;)

fig.fdr &lt;- (fig.fdr.a + fig.fdr.b) / (fig.fdr.c + fig.fdr.d) +
  plot_annotation(tag_levels = &#39;a&#39;)

ggsave(plot = fig.fdr,
       filename = file.path(path.misc,&quot;figure4.pdf&quot;),
       device = &#39;pdf&#39;,width = 5,height = 5,units = &#39;in&#39;,dpi = 300)

fig.fdr</code></pre>
<p><img src="figure/simulation-paper.Rmd/simulation-paper_fdr-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-simulation-paper_fdr-1">
Past versions of simulation-paper_fdr-1.png
</button>
</p>
<div id="fig-simulation-paper_fdr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/623d4297221b6b8a0e0e44292e19626bf6ed3270/docs/figure/simulation-paper.Rmd/simulation-paper_fdr-1.png" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/5dcd60b1e32ad8f39163c4ac3727c17220cb5ad1/docs/figure/simulation-paper.Rmd/simulation-paper_fdr-1.png" target="_blank">5dcd60b</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/d34d4e689994129c6bbdc8a356fe5778afbdd1e1/docs/figure/simulation-paper.Rmd/simulation-paper_fdr-1.png" target="_blank">d34d4e6</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="type-1-error" class="section level2">
<h2>Type 1 error</h2>
<p>Figure 5 is created below.</p>
<pre class="r"><code>dt.type1error &lt;- rbind(subsetDT(dt.metrics,scenario.balanced,&#39;B&#39;),
                       subsetDT(dt.metrics,scenario.unbalanced,&#39;B&#39;))

dt.type1error &lt;- dt.type1error[LibsPerGroup != &quot;#Lib/Group = 10&quot;,]

dt.type1error$LibsPerGroup %&lt;&gt;% 
  mapvalues(from = paste0(&#39;#Lib/Group = &#39;, c(3, 5)),
            to = paste0(c(3,5),&#39; samples per group&#39;))

dt.type1error$Scenario %&lt;&gt;% 
  mapvalues(from = c(&#39;balanced&#39;,&#39;unbalanced&#39;),
            to = c(&#39;Equal library sizes&#39;,&#39;Unequal library sizes&#39;))

dt.type1error[, FDR := roundPretty(ifelse((FP+TP) == 0,NA,100*FP/(FP+TP)),1)]

dt.type1error &lt;- dt.type1error[TxPerGene == &#39;All Transcripts&#39;,]


sub.byvar &lt;- 
  colnames(dt.type1error)[-which(colnames(dt.type1error) %in% c(&#39;P.SIG&#39;,&#39;TP&#39;,&#39;FP&#39;))]

x.melt &lt;- 
  melt(dt.type1error,id.vars = sub.byvar,
       measure.vars = c(&#39;P.SIG&#39;),variable.name = &#39;Type&#39;,value.name = &#39;Value&#39;)

plot.type1error &lt;- function(df.bar,scenario,library,legend = FALSE,base_size = 8){
  tb.bar &lt;- df.bar[Scenario == scenario &amp; LibsPerGroup == library,]
  
  ggplot(tb.bar,aes(x = Method,y = Value)) +
    geom_col(fill = &quot;#bebebe&quot;,col = &#39;black&#39;) +
    geom_hline(yintercept = 0.05,color = &#39;#ff0000&#39;,linetype = &#39;dashed&#39;,linewidth = 0.5) +
    labs(x = NULL,y = paste(&#39;Type 1 error rate&#39;)) +
    scale_y_continuous(limits = c(0,0.06),breaks = c(0,0.02,0.04,0.06)) +
    theme_bw(base_size = base_size,base_family = &#39;sans&#39;) +
    theme(panel.grid = element_blank(),
          axis.text.x = element_text(angle = 90),
          axis.text = element_text(colour = &#39;black&#39;,size = base_size))
}

fig.type1error.a &lt;- plot.type1error(df.bar = x.melt,scenario = &#39;Equal library sizes&#39;,library = &#39;3 samples per group&#39;)
fig.type1error.b &lt;- plot.type1error(df.bar = x.melt,scenario = &#39;Unequal library sizes&#39;,library = &#39;3 samples per group&#39;)
fig.type1error.c &lt;- plot.type1error(df.bar = x.melt,scenario = &#39;Equal library sizes&#39;,library = &#39;5 samples per group&#39;)
fig.type1error.d &lt;- plot.type1error(df.bar = x.melt,scenario = &#39;Unequal library sizes&#39;,library = &#39;5 samples per group&#39;)

fig.type1error &lt;- (fig.type1error.a + fig.type1error.b) / (fig.type1error.c + fig.type1error.d) +
  plot_annotation(tag_levels = &#39;a&#39;)

ggsave(plot = fig.type1error,
       filename = file.path(path.misc,&quot;figure5.pdf&quot;),
       device = &#39;pdf&#39;,width = 5,height = 5,units = &#39;in&#39;,dpi = 300)

fig.type1error</code></pre>
<p><img src="figure/simulation-paper.Rmd/simulation-paper_type1error_barplot-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-simulation-paper_type1error_barplot-1">
Past versions of simulation-paper_type1error_barplot-1.png
</button>
</p>
<div id="fig-simulation-paper_type1error_barplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/623d4297221b6b8a0e0e44292e19626bf6ed3270/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_barplot-1.png" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/49c9a940256334199278c62ffc22b49d22fa6883/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_barplot-1.png" target="_blank">49c9a94</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/5dcd60b1e32ad8f39163c4ac3727c17220cb5ad1/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_barplot-1.png" target="_blank">5dcd60b</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/d34d4e689994129c6bbdc8a356fe5778afbdd1e1/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_barplot-1.png" target="_blank">d34d4e6</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, we generate Figure 6.</p>
<pre class="r"><code>dt.pvalue.plot &lt;- subsetDT(dt.pvalue,scenario.unbalanced,&#39;C&#39;,&#39;All Transcripts&#39;)
dt.pvalue.plot &lt;- dt.pvalue.plot[LibsPerGroup == &#39;#Lib/Group = 5&#39;,]

plot.hist &lt;- function(df.hist,method,legend = FALSE,base_size = 8){
  tb.bar &lt;- df.hist[Method == method,]
  
  ggplot(data = tb.bar,aes(x = PValue,y = Density.Avg)) +
    geom_col(fill = &quot;#bebebe&quot;,col = &#39;black&#39;,position = position_dodge(),width = 0.75) +
    geom_hline(yintercept = 1,col = &#39;#ff0000&#39;,linetype = &#39;dashed&#39;,linewidth = 0.5) +
    scale_x_discrete(breaks = c(&quot;(0.00-0.05]&quot;,&quot;(0.50-0.55]&quot;,&quot;(0.95-1.00]&quot;),
                     labels = c(0.00,0.50,1.00)) +
    labs(x = &#39;P-values&#39;,y = &#39;Density&#39;) +
    theme_bw(base_size = base_size,base_family = &#39;sans&#39;) +
    theme(panel.grid = element_blank(),
          axis.text = element_text(colour = &#39;black&#39;,size = base_size))
}

fig.hist.a &lt;- plot.hist(df.hist = dt.pvalue.plot,method = &#39;edgeR-raw&#39;)
fig.hist.b &lt;- plot.hist(df.hist = dt.pvalue.plot,method = &#39;edgeR-scaled&#39;)
fig.hist.c &lt;- plot.hist(df.hist = dt.pvalue.plot,method = &#39;sleuth-LRT&#39;)
fig.hist.d &lt;- plot.hist(df.hist = dt.pvalue.plot,method = &#39;sleuth-Wald&#39;)
fig.hist.e &lt;- plot.hist(df.hist = dt.pvalue.plot,method = &#39;Swish&#39;)

design &lt;- c(
  area(1,1),area(1,2),
  area(2,1),area(2,2),
  area(3,1)
)

fig.hist &lt;- fig.hist.a + fig.hist.b + fig.hist.c + fig.hist.d + fig.hist.e +
  plot_layout(design = design) +
  plot_annotation(tag_levels = &#39;a&#39;)

ggsave(plot = fig.hist,
       filename = file.path(path.misc,&quot;figure6.pdf&quot;),
       device = &#39;pdf&#39;,width = 5,height = 7.5,units = &#39;in&#39;,dpi = 300)

fig.hist</code></pre>
<p><img src="figure/simulation-paper.Rmd/simulation-paper_type1error_pvalue-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-simulation-paper_type1error_pvalue-1">
Past versions of simulation-paper_type1error_pvalue-1.png
</button>
</p>
<div id="fig-simulation-paper_type1error_pvalue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/6a33f36051b3cc3197e6702589169bd914f98d04/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_pvalue-1.png" target="_blank">6a33f36</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-02-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/623d4297221b6b8a0e0e44292e19626bf6ed3270/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_pvalue-1.png" target="_blank">623d429</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/49c9a940256334199278c62ffc22b49d22fa6883/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_pvalue-1.png" target="_blank">49c9a94</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/5dcd60b1e32ad8f39163c4ac3727c17220cb5ad1/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_pvalue-1.png" target="_blank">5dcd60b</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2023-01-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/plbaldoni/TranscriptDE-code/blob/d34d4e689994129c6bbdc8a356fe5778afbdd1e1/docs/figure/simulation-paper.Rmd/simulation-paper_type1error_pvalue-1.png" target="_blank">d34d4e6</a>
</td>
<td>
Pedro Baldoni
</td>
<td>
2022-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="read-type" class="section level2">
<h2>Read type</h2>
<p>Below we generate Table 1 of the main paper.</p>
<pre class="r"><code># Overdispersion fold-change
dt.sigma2 &lt;- dt.overdispersion[TxPerGene == &#39;All Transcripts&#39; &amp; 
                                 Quantifier == &#39;Salmon&#39; &amp; 
                                 Scenario == &#39;unbalanced&#39; &amp; 
                                 FC == &#39;fc2&#39; &amp;
                                 LibsPerGroup != &#39;#Lib/Group = 10&#39;,]

dt.sigma2 &lt;- dt.sigma2[,-c(1,3,5,6,8,10:15)]

dt.sigma2.150.PE &lt;- dt.sigma2[Length == &#39;150bp&#39; &amp; Reads == &#39;paired-end&#39;,][,-c(1,2)]
setnames(dt.sigma2.150.PE,old = &#39;Mean&#39;,new = &#39;Mean.150.PE&#39;)

dt.sigma2 &lt;- merge(dt.sigma2,dt.sigma2.150.PE,by = c(&#39;LibsPerGroup&#39;),
                   all.x=TRUE,sort = FALSE)

dt.sigma2[,FC := Mean - Mean.150.PE]

dt.sigma2.3 &lt;- dt.sigma2[LibsPerGroup == &#39;#Lib/Group = 3&#39;,]
dt.sigma2.5 &lt;- dt.sigma2[LibsPerGroup == &#39;#Lib/Group = 5&#39;,]

dt.sigma2.3 &lt;- dcast(dt.sigma2.3,LibsPerGroup + Length ~ Reads,value.var = &#39;FC&#39;)
dt.sigma2.5 &lt;- dcast(dt.sigma2.5,LibsPerGroup + Length ~ Reads,value.var = &#39;FC&#39;)

setnames(dt.sigma2.3,
         old = c(&#39;paired-end&#39;,&#39;single-end&#39;),
         new = c(&#39;FC.PE&#39;,&#39;FC.SE&#39;))
setnames(dt.sigma2.5,
         old = c(&#39;paired-end&#39;,&#39;single-end&#39;),
         new = c(&#39;FC.PE&#39;,&#39;FC.SE&#39;))

setcolorder(dt.sigma2.3,neworder = c(&#39;LibsPerGroup&#39;,&#39;Length&#39;,&#39;FC.PE&#39;,&#39;FC.SE&#39;))
setcolorder(dt.sigma2.5,neworder = c(&#39;LibsPerGroup&#39;,&#39;Length&#39;,&#39;FC.PE&#39;,&#39;FC.SE&#39;))

dt.sigma2.long &lt;- rbind(dt.sigma2.3,dt.sigma2.5)
dt.sigma2.long$LibsPerGroup %&lt;&gt;% 
  mapvalues(from = c(&#39;#Lib/Group = 3&#39;,&#39;#Lib/Group = 5&#39;),to = c(3,5))
dt.sigma2.long$Length %&lt;&gt;% factor(levels = paste0(seq(50,150,25),&#39;bp&#39;))
dt.sigma2.long &lt;- dt.sigma2.long[order(LibsPerGroup,Length),]

# Power and FDR
dt.scenario.table &lt;- 
  expand.grid(&#39;genome&#39; = &#39;mm39&#39;,
              &#39;quantifier&#39; = c(&#39;Salmon&#39;,&#39;kallisto&#39;),
              &#39;txpergene&#39; = c(paste0(&#39;#Tx/Gene = &#39;,2:5),&#39;All Transcripts&#39;),
              stringsAsFactors = FALSE)

dt.scenario.table &lt;- as.data.table(dt.scenario.table)

scenario.table &lt;- 
  dt.scenario.table[quantifier == &#39;Salmon&#39; &amp; txpergene == &#39;All Transcripts&#39;,]
scenario.table &lt;- as.character(scenario.table)
names(scenario.table) &lt;- colnames(dt.scenario.table)

dt.table &lt;- subsetDT(dt.metrics,scenario = scenario.table,plot = FALSE)
dt.table &lt;- dt.table[Method == &#39;edgeR-scaled&#39; &amp;
                       Scenario == &#39;unbalanced&#39; &amp;
                       LibsPerGroup != &#39;#Lib/Group = 10&#39;,]

dt.table[,Power := TP/3000]
dt.table[,FDR := ifelse((FP+TP) == 0,NA,FP/(FP+TP))]

dt.table.3 &lt;- dt.table[LibsPerGroup == &#39;#Lib/Group = 3&#39;,][,-c(1,3,5,6,8:12)]
dt.table.5 &lt;- dt.table[LibsPerGroup == &#39;#Lib/Group = 5&#39;,][,-c(1,3,5,6,8:12)]

dt.table.3 &lt;- dcast(dt.table.3,LibsPerGroup + Length ~ Reads,value.var = c(&#39;Power&#39;,&#39;FDR&#39;))
dt.table.5 &lt;- dcast(dt.table.5,LibsPerGroup + Length ~ Reads,value.var = c(&#39;Power&#39;,&#39;FDR&#39;))

setnames(dt.table.3,
         old = c(&#39;Power_paired-end&#39;,&#39;Power_single-end&#39;,&#39;FDR_paired-end&#39;,&#39;FDR_single-end&#39;),
         new = c(&#39;Power.PE&#39;,&#39;Power.SE&#39;,&#39;FDR.PE&#39;,&#39;FDR.SE&#39;))
setnames(dt.table.5,
         old = c(&#39;Power_paired-end&#39;,&#39;Power_single-end&#39;,&#39;FDR_paired-end&#39;,&#39;FDR_single-end&#39;),
         new = c(&#39;Power.PE&#39;,&#39;Power.SE&#39;,&#39;FDR.PE&#39;,&#39;FDR.SE&#39;))

setcolorder(dt.table.3,neworder = c(&#39;LibsPerGroup&#39;,&#39;Length&#39;,&#39;Power.SE&#39;,&#39;FDR.SE&#39;,&#39;Power.PE&#39;,&#39;FDR.PE&#39;))
setcolorder(dt.table.5,neworder = c(&#39;LibsPerGroup&#39;,&#39;Length&#39;,&#39;Power.SE&#39;,&#39;FDR.SE&#39;,&#39;Power.PE&#39;,&#39;FDR.PE&#39;))

dt.table.long &lt;- rbind(dt.table.3,dt.table.5)
dt.table.long$LibsPerGroup %&lt;&gt;% mapvalues(from = c(&#39;#Lib/Group = 3&#39;,&#39;#Lib/Group = 5&#39;),to = c(3,5))
dt.table.long$Length %&lt;&gt;% factor(levels = paste0(seq(50,150,25),&#39;bp&#39;))
dt.table.long &lt;- dt.table.long[order(LibsPerGroup,Length),]

# Organizing tables

dt.table.sigma2 &lt;- 
  merge(dt.table.long,dt.sigma2.long,
        all.x = TRUE,by = c(&#39;LibsPerGroup&#39;,&#39;Length&#39;),sort = FALSE)

setcolorder(dt.table.sigma2,
            neworder = c(&#39;LibsPerGroup&#39;,&#39;Length&#39;,
                         &#39;FC.SE&#39;,&#39;Power.SE&#39;,&#39;FDR.SE&#39;,
                         &#39;FC.PE&#39;,&#39;Power.PE&#39;,&#39;FDR.PE&#39;))

dt.table.sigma2[,Length := gsub(&#39;bp&#39;,&#39;&#39;,Length)]
dt.table.sigma2$LibsPerGroup %&lt;&gt;% mapvalues(from = c(3,5),to = c(&#39;Three&#39;,&#39;Five&#39;))

tb &lt;- kbl(dt.table.sigma2,digits = 3,format = &#39;latex&#39;,escape = FALSE,booktabs = TRUE,
          align = c(&#39;c&#39;,&#39;r&#39;,rep(&#39;r&#39;,6)),
          col.names = linebreak(c(&#39;Samples per\ngroup&#39;,&#39;Read Length\n(bp)&#39;,
                                  &#39;Mapping Ambiguity\nlog-FC&#39;,&#39;Power&#39;,&#39;FDR&#39;,
                                  &#39;Mapping Ambiguity\nlog-FC&#39;,&#39;Power&#39;,&#39;FDR&#39;),align = &quot;c&quot;)) %&gt;%
  add_header_above(c(&quot; &quot; = 2, &quot;Single-end Read&quot; = 3, &quot;Paired-end Read&quot; = 3)) %&gt;%
  collapse_rows(1, latex_hline = &#39;major&#39;)

save_kable(tb,file = file.path(path.misc,&quot;table1.tex&quot;))</code></pre>
</div>
<div id="speed" class="section level2">
<h2>Speed</h2>
<p>In the main paper we also comment on methods’ performance regarding
computing time. The table below present such numbers.</p>
<pre class="r"><code>dt.time[, .(min =60*min(Time),
            mean = 60*mean(Time),
            med = 60*median(Time),
          max = 60*max(Time)),by = c(&#39;Quantifier&#39;,&#39;LibsPerGroup&#39;,&#39;Method&#39;)]</code></pre>
<pre><code>    Quantifier    LibsPerGroup       Method      min       mean        med
 1:   kallisto #Lib/Group = 10   sleuth-LRT 82.17615 121.016101 120.034750
 2:   kallisto #Lib/Group = 10  sleuth-Wald 70.17770  94.505086  94.504450
 3:   kallisto #Lib/Group = 10        Swish 82.49180 107.369005 108.558900
 4:   kallisto #Lib/Group = 10 edgeR-scaled 18.42285  24.523588  24.777975
 5:   kallisto #Lib/Group = 10    edgeR-raw 18.74970  25.577671  25.852075
 6:     Salmon #Lib/Group = 10   sleuth-LRT 88.06350 129.512314 130.150425
 7:     Salmon #Lib/Group = 10  sleuth-Wald 74.41825 100.213976 100.572475
 8:     Salmon #Lib/Group = 10        Swish 74.73545  96.503472  97.155575
 9:     Salmon #Lib/Group = 10 edgeR-scaled 12.73215  16.662010  16.812175
10:     Salmon #Lib/Group = 10    edgeR-raw 13.02015  17.524209  17.723825
11:   kallisto  #Lib/Group = 3   sleuth-LRT 48.90445 114.777883 117.900825
12:   kallisto  #Lib/Group = 3  sleuth-Wald 34.58840  85.275455  87.065625
13:   kallisto  #Lib/Group = 3        Swish 34.37065  45.465593  44.626050
14:   kallisto  #Lib/Group = 3 edgeR-scaled  7.03775   9.823749   9.811000
15:   kallisto  #Lib/Group = 3    edgeR-raw  7.41345  10.634582  10.534025
16:     Salmon  #Lib/Group = 3   sleuth-LRT 45.96170 112.039833 113.554825
17:     Salmon  #Lib/Group = 3  sleuth-Wald 31.45615  81.714932  83.387850
18:     Salmon  #Lib/Group = 3        Swish 31.83835  42.791656  42.444400
19:     Salmon  #Lib/Group = 3 edgeR-scaled  5.03500   6.981560   6.968950
20:     Salmon  #Lib/Group = 3    edgeR-raw  5.34300   7.697310   7.773125
21:   kallisto  #Lib/Group = 5   sleuth-LRT 61.24195 150.971167 155.360425
22:   kallisto  #Lib/Group = 5  sleuth-Wald 46.10565 121.186748 123.230125
23:   kallisto  #Lib/Group = 5        Swish 50.51140  64.607524  64.502500
24:   kallisto  #Lib/Group = 5 edgeR-scaled 10.49230  14.251279  14.349650
25:   kallisto  #Lib/Group = 5    edgeR-raw 10.87865  15.045484  15.178575
26:     Salmon  #Lib/Group = 5   sleuth-LRT 55.51295 149.997005 154.280775
27:     Salmon  #Lib/Group = 5  sleuth-Wald 41.56545 119.838872 120.990575
28:     Salmon  #Lib/Group = 5        Swish 46.06595  59.589219  59.646250
29:     Salmon  #Lib/Group = 5 edgeR-scaled  7.21000   9.741031   9.840850
30:     Salmon  #Lib/Group = 5    edgeR-raw  7.54285  10.564921  10.732775
    Quantifier    LibsPerGroup       Method      min       mean        med
          max
 1: 162.29385
 2: 118.90330
 3: 129.86270
 4:  29.97945
 5:  31.75405
 6: 176.43940
 7: 131.24655
 8: 120.01060
 9:  21.74150
10:  22.03945
11: 239.02190
12: 163.38015
13:  64.40590
14:  14.38945
15:  18.12705
16: 291.38200
17: 148.87335
18:  85.81485
19:   9.84275
20:  10.67450
21: 248.24995
22: 203.70500
23:  81.35545
24:  18.53090
25:  19.54600
26: 247.50205
27: 199.06345
28:  77.99000
29:  12.95745
30:  13.80970
          max</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /stornext/System/data/apps/R/R-4.3.0/lib64/R/lib/libRblas.so 
LAPACK: /stornext/System/data/apps/R/R-4.3.0/lib64/R/lib/libRlapack.so;  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pkg_1.0             ragg_1.2.5          patchwork_1.1.2    
 [4] kableExtra_1.3.4    ggpubr_0.6.0        readr_2.1.4        
 [7] purrr_1.0.1         devtools_2.4.5      usethis_2.2.2      
[10] BiocParallel_1.34.2 edgeR_3.42.4        limma_3.56.2       
[13] magrittr_2.0.3      plyr_1.8.8          thematic_0.1.2.1   
[16] ggplot2_3.4.2       data.table_1.14.8   workflowr_1.7.0    

loaded via a namespace (and not attached):
  [1] splines_4.3.0                 later_1.3.1                  
  [3] BiocIO_1.10.0                 bitops_1.0-7                 
  [5] filelock_1.0.2                R.oo_1.25.0                  
  [7] tibble_3.2.1                  XML_3.99-0.14                
  [9] lifecycle_1.0.3               rstatix_0.7.2                
 [11] rprojroot_2.0.3               ensembldb_2.24.0             
 [13] processx_3.8.2                lattice_0.21-8               
 [15] backports_1.4.1               sass_0.4.6                   
 [17] rmarkdown_2.23                jquerylib_0.1.4              
 [19] yaml_2.3.7                    remotes_2.4.2                
 [21] httpuv_1.6.11                 sessioninfo_1.2.2            
 [23] pkgbuild_1.4.2                DBI_1.1.3                    
 [25] abind_1.4-5                   pkgload_1.3.2.1              
 [27] zlibbioc_1.46.0               rvest_1.0.3                  
 [29] GenomicRanges_1.52.0          R.utils_2.12.2               
 [31] AnnotationFilter_1.24.0       BiocGenerics_0.46.0          
 [33] RCurl_1.98-1.12               rappdirs_0.3.3               
 [35] git2r_0.32.0                  GenomeInfoDbData_1.2.10      
 [37] wasabi_1.0.1                  IRanges_2.34.1               
 [39] S4Vectors_0.38.1              fishpond_2.6.2               
 [41] svglite_2.1.1                 DelayedArray_0.26.6          
 [43] codetools_0.2-19              xml2_1.3.5                   
 [45] tidyselect_1.2.0              farver_2.1.1                 
 [47] matrixStats_1.0.0             stats4_4.3.0                 
 [49] BiocFileCache_2.8.0           webshot_0.5.5                
 [51] showtext_0.9-6                GenomicAlignments_1.36.0     
 [53] jsonlite_1.8.7                ellipsis_0.3.2               
 [55] systemfonts_1.0.4             tools_4.3.0                  
 [57] progress_1.2.2                Rcpp_1.0.11                  
 [59] glue_1.6.2                    svMisc_1.2.3                 
 [61] xfun_0.39                     MatrixGenerics_1.12.2        
 [63] GenomeInfoDb_1.36.1           dplyr_1.1.2                  
 [65] withr_2.5.0                   BiocManager_1.30.21          
 [67] fastmap_1.1.1                 rhdf5filters_1.12.1          
 [69] fansi_1.0.4                   callr_3.7.3                  
 [71] digest_0.6.33                 R6_2.5.1                     
 [73] mime_0.12                     textshaping_0.3.6            
 [75] colorspace_2.1-0              gtools_3.9.4                 
 [77] biomaRt_2.56.1                RSQLite_2.3.1                
 [79] R.methodsS3_1.8.2             utf8_1.2.3                   
 [81] tidyr_1.3.0                   generics_0.1.3               
 [83] tximeta_1.18.0                rtracklayer_1.60.0           
 [85] S4Arrays_1.0.4                prettyunits_1.1.1            
 [87] httr_1.4.6                    htmlwidgets_1.6.2            
 [89] whisker_0.4.1                 pkgconfig_2.0.3              
 [91] gtable_0.3.3                  blob_1.2.4                   
 [93] SingleCellExperiment_1.22.0   XVector_0.40.0               
 [95] htmltools_0.5.5               carData_3.0-5                
 [97] sysfonts_0.8.8                profvis_0.3.8                
 [99] ProtGenerics_1.32.0           sleuth_0.30.0                
[101] scales_1.2.1                  Biobase_2.60.0               
[103] Rsubread_2.14.2               png_0.1-8                    
[105] knitr_1.43                    rstudioapi_0.15.0            
[107] tzdb_0.4.0                    rjson_0.2.21                 
[109] curl_5.0.1                    showtextdb_3.0               
[111] cachem_1.0.8                  rhdf5_2.44.0                 
[113] stringr_1.5.0                 BiocVersion_3.17.1           
[115] parallel_4.3.0                miniUI_0.1.1.1               
[117] AnnotationDbi_1.62.2          restfulr_0.0.15              
[119] desc_1.4.2                    pillar_1.9.0                 
[121] grid_4.3.0                    vctrs_0.6.3                  
[123] urlchecker_1.0.1              promises_1.2.0.1             
[125] car_3.1-2                     dbplyr_2.3.3                 
[127] xtable_1.8-4                  tximport_1.28.0              
[129] evaluate_0.21                 GenomicFeatures_1.52.1       
[131] Rsamtools_2.16.0              cli_3.6.1                    
[133] locfit_1.5-9.8                compiler_4.3.0               
[135] rlang_1.1.1                   crayon_1.5.2                 
[137] ggsignif_0.6.4                labeling_0.4.2               
[139] ps_1.7.5                      getPass_0.2-2                
[141] fs_1.6.2                      stringi_1.7.12               
[143] viridisLite_0.4.2             munsell_0.5.0                
[145] Biostrings_2.68.1             lazyeval_0.2.2               
[147] Matrix_1.6-0                  hms_1.1.3                    
[149] bit64_4.0.5                   Rhdf5lib_1.22.0              
[151] KEGGREST_1.40.0               shiny_1.7.4.1                
[153] highr_0.10                    SummarizedExperiment_1.30.2  
[155] interactiveDisplayBase_1.38.0 AnnotationHub_3.8.0          
[157] broom_1.0.5                   memoise_2.0.1                
[159] bslib_0.5.0                   bit_4.0.5                    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
